// igniteui-angular/esm2015/lib/dialog/dialog.directives.js
import { Directive, HostBinding } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxDialogTitleDirective {
    constructor() {
        this.defaultStyle = true;
    }
}
IgxDialogTitleDirective.ɵfac = function IgxDialogTitleDirective_Factory(t) { return new (t || IgxDialogTitleDirective)(); };
IgxDialogTitleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxDialogTitleDirective, selectors: [["igx-dialog-title"], ["", "igxDialogTitle", ""]], hostVars: 2, hostBindings: function IgxDialogTitleDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-dialog__window-title", ctx.defaultStyle);
    } } });
IgxDialogTitleDirective.propDecorators = {
    defaultStyle: [{ type: HostBinding, args: ['class.igx-dialog__window-title',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxDialogTitleDirective, [{
        type: Directive,
        args: [{
                selector: 'igx-dialog-title,[igxDialogTitle]'
            }]
    }], function () { return []; }, { defaultStyle: [{
            type: HostBinding,
            args: ['class.igx-dialog__window-title']
        }] }); })();
/**
 * @hidden
 */
export class IgxDialogActionsDirective {
    constructor() {
        this.defaultClass = true;
    }
}
IgxDialogActionsDirective.ɵfac = function IgxDialogActionsDirective_Factory(t) { return new (t || IgxDialogActionsDirective)(); };
IgxDialogActionsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxDialogActionsDirective, selectors: [["igx-dialog-actions"], ["", "igxDialogActions", ""]], hostVars: 2, hostBindings: function IgxDialogActionsDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-dialog__window-actions", ctx.defaultClass);
    } } });
IgxDialogActionsDirective.propDecorators = {
    defaultClass: [{ type: HostBinding, args: ['class.igx-dialog__window-actions',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxDialogActionsDirective, [{
        type: Directive,
        args: [{
                selector: 'igx-dialog-actions,[igxDialogActions]'
            }]
    }], function () { return []; }, { defaultClass: [{
            type: HostBinding,
            args: ['class.igx-dialog__window-actions']
        }] }); })();
