// igniteui-angular/esm2015/lib/grids/headers/headers.module.js
import { NgModule } from '@angular/core';
import { IgxGridHeaderComponent } from './grid-header.component';
import { IgxGridHeaderGroupComponent } from './grid-header-group.component';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnMovingModule } from '../moving/moving.module';
import { IgxGridFilteringModule } from '../filtering/base/filtering.module';
import { IgxGridResizingModule } from '../resizing/resize.module';
import { SortingIndexPipe } from './sorting-index.pipe';
import * as ɵngcc0 from '@angular/core';
export class IgxGridHeadersModule {
}
IgxGridHeadersModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridHeadersModule });
IgxGridHeadersModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridHeadersModule_Factory(t) { return new (t || IgxGridHeadersModule)(); }, imports: [[
            IgxGridSharedModules,
            IgxGridFilteringModule,
            IgxColumnMovingModule,
            IgxGridResizingModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridHeadersModule, { declarations: function () { return [IgxGridHeaderComponent,
        IgxGridHeaderGroupComponent,
        SortingIndexPipe]; }, imports: function () { return [IgxGridSharedModules,
        IgxGridFilteringModule,
        IgxColumnMovingModule,
        IgxGridResizingModule]; }, exports: function () { return [IgxGridHeaderComponent,
        IgxGridHeaderGroupComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridHeadersModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxGridHeaderComponent,
                    IgxGridHeaderGroupComponent,
                    SortingIndexPipe
                ],
                imports: [
                    IgxGridSharedModules,
                    IgxGridFilteringModule,
                    IgxColumnMovingModule,
                    IgxGridResizingModule
                ],
                exports: [
                    IgxGridHeaderComponent,
                    IgxGridHeaderGroupComponent
                ]
            }]
    }], null, null); })();
