// igniteui-angular/esm2015/lib/select/select-group.component.js
import { Component } from '@angular/core';
import { IgxDropDownGroupComponent } from '../drop-down/public_api';
/**
 * The `<igx-select-item>` is a container intended for row items in
 * a `<igx-select>` container.
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = [[["igx-select-item"]]];
const _c1 = ["igx-select-item"];
export class IgxSelectGroupComponent extends IgxDropDownGroupComponent {
}
IgxSelectGroupComponent.ɵfac = function IgxSelectGroupComponent_Factory(t) { return ɵIgxSelectGroupComponent_BaseFactory(t || IgxSelectGroupComponent); };
IgxSelectGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxSelectGroupComponent, selectors: [["igx-select-item-group"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c1, decls: 3, vars: 2, consts: [[3, "id"]], template: function IgxSelectGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "label", 0);
        ɵngcc0.ɵɵtext(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(2);
    } if (rf & 2) {
        ɵngcc0.ɵɵpropertyInterpolate("id", ctx.labelId);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ctx.label);
    } }, encapsulation: 2 });
const ɵIgxSelectGroupComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(IgxSelectGroupComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxSelectGroupComponent, [{
        type: Component,
        args: [{
                selector: 'igx-select-item-group',
                template: `
        <label id="{{labelId}}">{{ label }}</label>
        <ng-content select="igx-select-item"></ng-content>
    `
            }]
    }], null, null); })();
