// igniteui-angular/esm2015/lib/carousel/carousel.directives.js
import { Directive } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class IgxCarouselIndicatorDirective {
}
IgxCarouselIndicatorDirective.ɵfac = function IgxCarouselIndicatorDirective_Factory(t) { return new (t || IgxCarouselIndicatorDirective)(); };
IgxCarouselIndicatorDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCarouselIndicatorDirective, selectors: [["", "igxCarouselIndicator", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCarouselIndicatorDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCarouselIndicator]'
            }]
    }], null, null); })();
export class IgxCarouselNextButtonDirective {
}
IgxCarouselNextButtonDirective.ɵfac = function IgxCarouselNextButtonDirective_Factory(t) { return new (t || IgxCarouselNextButtonDirective)(); };
IgxCarouselNextButtonDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCarouselNextButtonDirective, selectors: [["", "igxCarouselNextButton", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCarouselNextButtonDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCarouselNextButton]'
            }]
    }], null, null); })();
export class IgxCarouselPrevButtonDirective {
}
IgxCarouselPrevButtonDirective.ɵfac = function IgxCarouselPrevButtonDirective_Factory(t) { return new (t || IgxCarouselPrevButtonDirective)(); };
IgxCarouselPrevButtonDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCarouselPrevButtonDirective, selectors: [["", "igxCarouselPrevButton", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCarouselPrevButtonDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCarouselPrevButton]'
            }]
    }], null, null); })();
