// igniteui-angular/esm2015/lib/grids/tree-grid/tree-grid.directives.js
import { Directive, TemplateRef } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxRowLoadingIndicatorTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxRowLoadingIndicatorTemplateDirective.ɵfac = function IgxRowLoadingIndicatorTemplateDirective_Factory(t) { return new (t || IgxRowLoadingIndicatorTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxRowLoadingIndicatorTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxRowLoadingIndicatorTemplateDirective, selectors: [["", "igxRowLoadingIndicator", ""]] });
IgxRowLoadingIndicatorTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxRowLoadingIndicatorTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxRowLoadingIndicator]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
