// igniteui-angular/esm2015/lib/grids/grid/grid.details.pipe.js
import { Pipe } from '@angular/core';
import { GridBaseAPIService } from '../api.service';
/** @hidden */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../api.service';
export class IgxGridDetailsPipe {
    constructor(gridAPI) {
        this.gridAPI = gridAPI;
    }
    transform(collection, hasDetails, expansionStates, pipeTrigger) {
        if (!hasDetails) {
            return collection;
        }
        const res = this.addDetailRows(collection, expansionStates);
        return res;
    }
    addDetailRows(collection, expansionStates) {
        const result = [];
        collection.forEach((v) => {
            result.push(v);
            if (!this.gridAPI.grid.isGroupByRecord(v) && !this.gridAPI.grid.isSummaryRow(v) &&
                this.gridAPI.get_row_expansion_state(v)) {
                const detailsObj = { detailsData: v };
                result.push(detailsObj);
            }
        });
        return result;
    }
}
IgxGridDetailsPipe.ɵfac = function IgxGridDetailsPipe_Factory(t) { return new (t || IgxGridDetailsPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.GridBaseAPIService)); };
IgxGridDetailsPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "gridDetails", type: IgxGridDetailsPipe, pure: true });
IgxGridDetailsPipe.ctorParameters = () => [
    { type: GridBaseAPIService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridDetailsPipe, [{
        type: Pipe,
        args: [{
                name: 'gridDetails',
                pure: true
            }]
    }], function () { return [{ type: ɵngcc1.GridBaseAPIService }]; }, null); })();
