// igniteui-angular/esm2015/lib/grids/columns/templates.directive.js
import { Directive, TemplateRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class IgxFilterCellTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxFilterCellTemplateDirective.ɵfac = function IgxFilterCellTemplateDirective_Factory(t) { return new (t || IgxFilterCellTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxFilterCellTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxFilterCellTemplateDirective, selectors: [["", "igxFilterCellTemplate", ""]] });
IgxFilterCellTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxFilterCellTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxFilterCellTemplate]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
export class IgxCellTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxCellTemplateDirective.ɵfac = function IgxCellTemplateDirective_Factory(t) { return new (t || IgxCellTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxCellTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCellTemplateDirective, selectors: [["", "igxCell", ""]] });
IgxCellTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCellTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCell]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
export class IgxCellHeaderTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxCellHeaderTemplateDirective.ɵfac = function IgxCellHeaderTemplateDirective_Factory(t) { return new (t || IgxCellHeaderTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxCellHeaderTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCellHeaderTemplateDirective, selectors: [["", "igxHeader", ""]] });
IgxCellHeaderTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCellHeaderTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxHeader]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
/**
 * @hidden
 */
export class IgxCellFooterTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxCellFooterTemplateDirective.ɵfac = function IgxCellFooterTemplateDirective_Factory(t) { return new (t || IgxCellFooterTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxCellFooterTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCellFooterTemplateDirective, selectors: [["", "igxFooter", ""]] });
IgxCellFooterTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCellFooterTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxFooter]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
export class IgxCellEditorTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxCellEditorTemplateDirective.ɵfac = function IgxCellEditorTemplateDirective_Factory(t) { return new (t || IgxCellEditorTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxCellEditorTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCellEditorTemplateDirective, selectors: [["", "igxCellEditor", ""]] });
IgxCellEditorTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCellEditorTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCellEditor]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
export class IgxCollapsibleIndicatorTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxCollapsibleIndicatorTemplateDirective.ɵfac = function IgxCollapsibleIndicatorTemplateDirective_Factory(t) { return new (t || IgxCollapsibleIndicatorTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxCollapsibleIndicatorTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxCollapsibleIndicatorTemplateDirective, selectors: [["", "igxCollapsibleIndicator", ""]] });
IgxCollapsibleIndicatorTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCollapsibleIndicatorTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxCollapsibleIndicator]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
