// igniteui-angular/esm2015/lib/navigation-drawer/navigation-drawer.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxNavigationDrawerComponent } from './navigation-drawer.component';
import { IgxNavDrawerItemDirective, IgxNavDrawerMiniTemplateDirective, IgxNavDrawerTemplateDirective } from './navigation-drawer.directives';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxNavigationDrawerModule {
}
IgxNavigationDrawerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxNavigationDrawerModule });
IgxNavigationDrawerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxNavigationDrawerModule_Factory(t) { return new (t || IgxNavigationDrawerModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxNavigationDrawerModule, { declarations: function () { return [IgxNavigationDrawerComponent,
        IgxNavDrawerItemDirective,
        IgxNavDrawerMiniTemplateDirective,
        IgxNavDrawerTemplateDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [IgxNavigationDrawerComponent,
        IgxNavDrawerItemDirective,
        IgxNavDrawerMiniTemplateDirective,
        IgxNavDrawerTemplateDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxNavigationDrawerModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxNavigationDrawerComponent,
                    IgxNavDrawerItemDirective,
                    IgxNavDrawerMiniTemplateDirective,
                    IgxNavDrawerTemplateDirective
                ],
                exports: [
                    IgxNavigationDrawerComponent,
                    IgxNavDrawerItemDirective,
                    IgxNavDrawerMiniTemplateDirective,
                    IgxNavDrawerTemplateDirective
                ],
                imports: [CommonModule]
            }]
    }], null, null); })();
