// igniteui-angular/esm2015/lib/grids/summaries/grid-root-summary.pipe.js
import { Pipe } from '@angular/core';
import { GridBaseAPIService } from '../api.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../api.service';
export class IgxSummaryDataPipe {
    constructor(gridAPI) {
        this.gridAPI = gridAPI;
    }
    transform(id, trigger = 0) {
        const summaryService = this.gridAPI.grid.summaryService;
        return summaryService.calculateSummaries(summaryService.rootSummaryID, this.gridAPI.get_summary_data());
    }
}
IgxSummaryDataPipe.ɵfac = function IgxSummaryDataPipe_Factory(t) { return new (t || IgxSummaryDataPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.GridBaseAPIService)); };
IgxSummaryDataPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "igxGridSummaryDataPipe", type: IgxSummaryDataPipe, pure: true });
IgxSummaryDataPipe.ctorParameters = () => [
    { type: GridBaseAPIService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxSummaryDataPipe, [{
        type: Pipe,
        args: [{
                name: 'igxGridSummaryDataPipe',
                pure: true
            }]
    }], function () { return [{ type: ɵngcc1.GridBaseAPIService }]; }, null); })();
