// igniteui-angular/esm2015/lib/grids/summaries/summary.module.js
import { NgModule } from '@angular/core';
import { IgxSummaryCellComponent } from './summary-cell.component';
import { IgxSummaryRowComponent } from './summary-row.component';
import { IgxSummaryDataPipe } from './grid-root-summary.pipe';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxGridPipesModule } from '../common/grid-pipes.module';
import * as ɵngcc0 from '@angular/core';
export class IgxGridSummaryModule {
}
IgxGridSummaryModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridSummaryModule });
IgxGridSummaryModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridSummaryModule_Factory(t) { return new (t || IgxGridSummaryModule)(); }, imports: [[
            IgxGridPipesModule,
            IgxGridSharedModules
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridSummaryModule, { declarations: function () { return [IgxSummaryCellComponent,
        IgxSummaryRowComponent,
        IgxSummaryDataPipe]; }, imports: function () { return [IgxGridPipesModule,
        IgxGridSharedModules]; }, exports: function () { return [IgxSummaryCellComponent,
        IgxSummaryRowComponent,
        IgxSummaryDataPipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridSummaryModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxSummaryCellComponent,
                    IgxSummaryRowComponent,
                    IgxSummaryDataPipe
                ],
                imports: [
                    IgxGridPipesModule,
                    IgxGridSharedModules
                ],
                exports: [
                    IgxSummaryCellComponent,
                    IgxSummaryRowComponent,
                    IgxSummaryDataPipe
                ]
            }]
    }], null, null); })();
