// igniteui-angular/esm2015/lib/chips/chips.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { IgxRippleModule } from '../directives/ripple/ripple.directive';
import { IgxButtonModule } from '../directives/button/button.directive';
import { IgxAvatarModule } from '../avatar/avatar.component';
import { IgxIconModule } from '../icon/public_api';
import { IgxChipComponent } from './chip.component';
import { IgxChipsAreaComponent } from './chips-area.component';
import { IgxDragDropModule } from '../directives/drag-drop/drag-drop.directive';
import { IgxPrefixModule, IgxPrefixDirective } from '../directives/prefix/prefix.directive';
import { IgxSuffixModule, IgxSuffixDirective } from '../directives/suffix/suffix.directive';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxChipsModule {
}
IgxChipsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxChipsModule });
IgxChipsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxChipsModule_Factory(t) { return new (t || IgxChipsModule)(); }, imports: [[
            CommonModule,
            IgxRippleModule,
            IgxIconModule,
            IgxButtonModule,
            IgxAvatarModule,
            IgxDragDropModule,
            IgxPrefixModule,
            IgxSuffixModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxChipsModule, { declarations: function () { return [IgxChipsAreaComponent,
        IgxChipComponent]; }, imports: function () { return [CommonModule,
        IgxRippleModule,
        IgxIconModule,
        IgxButtonModule,
        IgxAvatarModule,
        IgxDragDropModule,
        IgxPrefixModule,
        IgxSuffixModule]; }, exports: function () { return [IgxChipsAreaComponent,
        IgxChipComponent,
        IgxPrefixDirective,
        IgxSuffixDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxChipsModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxChipsAreaComponent,
                    IgxChipComponent
                ],
                exports: [
                    IgxChipsAreaComponent,
                    IgxChipComponent,
                    IgxPrefixDirective,
                    IgxSuffixDirective
                ],
                imports: [
                    CommonModule,
                    IgxRippleModule,
                    IgxIconModule,
                    IgxButtonModule,
                    IgxAvatarModule,
                    IgxDragDropModule,
                    IgxPrefixModule,
                    IgxSuffixModule
                ]
            }]
    }], null, null); })();
