// igniteui-angular/esm2015/lib/grids/selection/row-selectors.js
import { Directive, TemplateRef } from '@angular/core';
/**
 * @hidden
 * @internal
 */
import * as ɵngcc0 from '@angular/core';
export class IgxRowSelectorDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
IgxRowSelectorDirective.ɵfac = function IgxRowSelectorDirective_Factory(t) { return new (t || IgxRowSelectorDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxRowSelectorDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxRowSelectorDirective, selectors: [["", "igxRowSelector", ""]] });
IgxRowSelectorDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxRowSelectorDirective, [{
        type: Directive,
        args: [{
                selector: '[igxRowSelector]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
/**
 * @hidden
 * @internal
 */
export class IgxHeadSelectorDirective {
    constructor(templateRef) {
        this.templateRef = templateRef;
    }
}
IgxHeadSelectorDirective.ɵfac = function IgxHeadSelectorDirective_Factory(t) { return new (t || IgxHeadSelectorDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxHeadSelectorDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxHeadSelectorDirective, selectors: [["", "igxHeadSelector", ""]] });
IgxHeadSelectorDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxHeadSelectorDirective, [{
        type: Directive,
        args: [{
                selector: '[igxHeadSelector]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
