// igniteui-angular/esm2015/lib/grids/selection/selection.module.js
import { NgModule } from '@angular/core';
import { IgxRowSelectorDirective, IgxHeadSelectorDirective } from './row-selectors';
import { IgxGridDragSelectDirective } from './drag-select.directive';
import * as ɵngcc0 from '@angular/core';
export class IgxGridSelectionModule {
}
IgxGridSelectionModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridSelectionModule });
IgxGridSelectionModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridSelectionModule_Factory(t) { return new (t || IgxGridSelectionModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridSelectionModule, { declarations: function () { return [IgxRowSelectorDirective,
        IgxHeadSelectorDirective,
        IgxGridDragSelectDirective]; }, exports: function () { return [IgxRowSelectorDirective,
        IgxHeadSelectorDirective,
        IgxGridDragSelectDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridSelectionModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxRowSelectorDirective,
                    IgxHeadSelectorDirective,
                    IgxGridDragSelectDirective
                ],
                exports: [
                    IgxRowSelectorDirective,
                    IgxHeadSelectorDirective,
                    IgxGridDragSelectDirective
                ]
            }]
    }], null, null); })();
