// igniteui-angular/esm2015/lib/grids/column-actions/column-actions.module.js
import { NgModule } from '@angular/core';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnHidingDirective } from './column-hiding.directive';
import { IgxColumnPinningDirective } from './column-pinning.directive';
import { IgxColumnActionsComponent } from './column-actions.component';
import { IgxGridPipesModule } from '../common/grid-pipes.module';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from '../../directives/button/button.directive';
import * as ɵngcc4 from '../../date-picker/date-picker.component';
import * as ɵngcc5 from '../../date-picker/date-picker.directives';
import * as ɵngcc6 from '../../icon/icon.component';
import * as ɵngcc7 from '../../directives/ripple/ripple.directive';
import * as ɵngcc8 from '../../input-group/input-group.component';
import * as ɵngcc9 from '../../directives/hint/hint.directive';
import * as ɵngcc10 from '../../directives/input/input.directive';
import * as ɵngcc11 from '../../directives/label/label.directive';
import * as ɵngcc12 from '../../directives/prefix/prefix.directive';
import * as ɵngcc13 from '../../directives/suffix/suffix.directive';
import * as ɵngcc14 from '../../directives/focus/focus.directive';
import * as ɵngcc15 from '../../directives/toggle/toggle.directive';
import * as ɵngcc16 from '../../directives/for-of/for_of.directive';
import * as ɵngcc17 from '../../directives/template-outlet/template_outlet.directive';
import * as ɵngcc18 from '../../directives/text-highlight/text-highlight.directive';
import * as ɵngcc19 from '../state.directive';
import * as ɵngcc20 from '../../directives/text-selection/text-selection.directive';
import * as ɵngcc21 from '../../checkbox/checkbox.component';
import * as ɵngcc22 from '../../badge/badge.component';
import * as ɵngcc23 from '../../chips/chips-area.component';
import * as ɵngcc24 from '../../chips/chip.component';
import * as ɵngcc25 from '../../directives/drag-drop/drag-drop.directive';
import * as ɵngcc26 from '../../drop-down/drop-down.component';
import * as ɵngcc27 from '../../drop-down/drop-down-item.component';
import * as ɵngcc28 from '../../drop-down/drop-down-group.component';
import * as ɵngcc29 from '../../drop-down/drop-down-navigation.directive';
import * as ɵngcc30 from '../../buttonGroup/buttonGroup.component';
import * as ɵngcc31 from '../../progressbar/progressbar.component';
import * as ɵngcc32 from '../../progressbar/progressbar.common';
import * as ɵngcc33 from '../../select/select.component';
import * as ɵngcc34 from '../../select/select-group.component';
import * as ɵngcc35 from '../../select/select-item.component';
import * as ɵngcc36 from '../../select/select-navigation.directive';
import * as ɵngcc37 from '../../snackbar/snackbar.component';
import * as ɵngcc38 from '../../date-picker/date-picker.pipes';
import * as ɵngcc39 from '../common/pipes';
export class IgxColumnActionsModule {
}
IgxColumnActionsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxColumnActionsModule });
IgxColumnActionsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxColumnActionsModule_Factory(t) { return new (t || IgxColumnActionsModule)(); }, imports: [[
            IgxGridSharedModules,
            IgxGridPipesModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxColumnActionsModule, { declarations: function () { return [IgxColumnHidingDirective,
        IgxColumnPinningDirective,
        IgxColumnActionsComponent]; }, imports: function () { return [IgxGridSharedModules,
        IgxGridPipesModule]; }, exports: function () { return [IgxColumnHidingDirective,
        IgxColumnPinningDirective,
        IgxColumnActionsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnActionsModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxColumnHidingDirective,
                    IgxColumnPinningDirective,
                    IgxColumnActionsComponent,
                ],
                imports: [
                    IgxGridSharedModules,
                    IgxGridPipesModule
                ],
                exports: [
                    IgxColumnHidingDirective,
                    IgxColumnPinningDirective,
                    IgxColumnActionsComponent
                ]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(IgxColumnActionsComponent, [ɵngcc1.NgClass, ɵngcc1.NgComponentOutlet, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgTemplateOutlet, ɵngcc1.NgStyle, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc1.NgPlural, ɵngcc1.NgPluralCase, ɵngcc2.ɵangular_packages_forms_forms_y, ɵngcc2.NgSelectOption, ɵngcc2.ɵangular_packages_forms_forms_x, ɵngcc2.DefaultValueAccessor, ɵngcc2.NumberValueAccessor, ɵngcc2.RangeValueAccessor, ɵngcc2.CheckboxControlValueAccessor, ɵngcc2.SelectControlValueAccessor, ɵngcc2.SelectMultipleControlValueAccessor, ɵngcc2.RadioControlValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgControlStatusGroup, ɵngcc2.RequiredValidator, ɵngcc2.MinLengthValidator, ɵngcc2.MaxLengthValidator, ɵngcc2.PatternValidator, ɵngcc2.CheckboxRequiredValidator, ɵngcc2.EmailValidator, ɵngcc2.NgModel, ɵngcc2.NgModelGroup, ɵngcc2.NgForm, ɵngcc3.IgxButtonDirective, ɵngcc4.IgxDatePickerComponent, ɵngcc5.IgxDatePickerTemplateDirective, ɵngcc5.IgxDatePickerActionsDirective, ɵngcc6.IgxIconComponent, ɵngcc7.IgxRippleDirective, ɵngcc8.IgxInputGroupComponent, ɵngcc9.IgxHintDirective, ɵngcc10.IgxInputDirective, ɵngcc11.IgxLabelDirective, ɵngcc12.IgxPrefixDirective, ɵngcc13.IgxSuffixDirective, ɵngcc14.IgxFocusDirective, ɵngcc15.IgxToggleDirective, ɵngcc15.IgxToggleActionDirective, ɵngcc15.IgxOverlayOutletDirective, ɵngcc16.IgxForOfDirective, ɵngcc16.IgxGridForOfDirective, ɵngcc17.IgxTemplateOutletDirective, ɵngcc18.IgxTextHighlightDirective, ɵngcc19.IgxGridStateDirective, ɵngcc20.IgxTextSelectionDirective, ɵngcc21.IgxCheckboxComponent, ɵngcc21.IgxCheckboxRequiredDirective, ɵngcc22.IgxBadgeComponent, ɵngcc23.IgxChipsAreaComponent, ɵngcc24.IgxChipComponent, ɵngcc25.IgxDragDirective, ɵngcc25.IgxDropDirective, ɵngcc25.IgxDragHandleDirective, ɵngcc25.IgxDragIgnoreDirective, ɵngcc26.IgxDropDownComponent, ɵngcc27.IgxDropDownItemComponent, ɵngcc28.IgxDropDownGroupComponent, ɵngcc29.IgxDropDownItemNavigationDirective, ɵngcc30.IgxButtonGroupComponent, ɵngcc31.IgxLinearProgressBarComponent, ɵngcc31.IgxCircularProgressBarComponent, ɵngcc32.IgxProcessBarTextTemplateDirective, ɵngcc32.IgxProgressBarGradientDirective, ɵngcc33.IgxSelectComponent, ɵngcc33.IgxSelectFooterDirective, ɵngcc34.IgxSelectGroupComponent, ɵngcc33.IgxSelectHeaderDirective, ɵngcc35.IgxSelectItemComponent, ɵngcc36.IgxSelectItemNavigationDirective, ɵngcc33.IgxSelectToggleIconDirective, ɵngcc37.IgxSnackbarComponent, IgxColumnHidingDirective,
    IgxColumnPinningDirective,
    IgxColumnActionsComponent], [ɵngcc1.AsyncPipe, ɵngcc1.UpperCasePipe, ɵngcc1.LowerCasePipe, ɵngcc1.JsonPipe, ɵngcc1.SlicePipe, ɵngcc1.DecimalPipe, ɵngcc1.PercentPipe, ɵngcc1.TitleCasePipe, ɵngcc1.CurrencyPipe, ɵngcc1.DatePipe, ɵngcc1.I18nPluralPipe, ɵngcc1.I18nSelectPipe, ɵngcc1.KeyValuePipe, ɵngcc38.DatePickerDisplayValuePipe, ɵngcc38.DatePickerInputValuePipe, ɵngcc39.IgxGridFilterConditionPipe, ɵngcc39.IgxGridTransactionPipe, ɵngcc39.IgxGridNotGroupedPipe, ɵngcc39.IgxGridTopLevelColumns, ɵngcc39.IgxGridCellStylesPipe, ɵngcc39.IgxGridCellStyleClassesPipe, ɵngcc39.IgxGridPaginatorOptionsPipe, ɵngcc39.IgxHasVisibleColumnsPipe, ɵngcc39.IgxGridRowPinningPipe, ɵngcc39.IgxColumnActionEnabledPipe, ɵngcc39.IgxFilterActionColumnsPipe, ɵngcc39.IgxSortActionColumnsPipe, ɵngcc39.IgxGridDataMapperPipe, ɵngcc39.IgxStringReplacePipe, ɵngcc39.IgxGridTransactionStatePipe, ɵngcc39.IgxGridAddRowPipe, ɵngcc39.IgxColumnFormatterPipe]);
