// igniteui-angular/esm2015/lib/core/navigation/nav.service.js
import { Injectable } from '@angular/core';
/**
 * Common service to be injected between components where those implementing common
 * ToggleView interface can register and toggle directives can call their methods.
 * TODO: Track currently active? Events?
 */
import * as ɵngcc0 from '@angular/core';
export class IgxNavigationService {
    constructor() {
        this.navs = {};
    }
    add(id, navItem) {
        this.navs[id] = navItem;
    }
    remove(id) {
        delete this.navs[id];
    }
    get(id) {
        if (id) {
            return this.navs[id];
        }
    }
    toggle(id, ...args) {
        if (this.navs[id]) {
            return this.navs[id].toggle(...args);
        }
    }
    open(id, ...args) {
        if (this.navs[id]) {
            return this.navs[id].open(...args);
        }
    }
    close(id, ...args) {
        if (this.navs[id]) {
            return this.navs[id].close(...args);
        }
    }
}
IgxNavigationService.ɵfac = function IgxNavigationService_Factory(t) { return new (t || IgxNavigationService)(); };
IgxNavigationService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: IgxNavigationService, factory: IgxNavigationService.ɵfac });
IgxNavigationService.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxNavigationService, [{
        type: Injectable
    }], function () { return []; }, null); })();
