// igniteui-angular/esm2015/lib/directives/for-of/display.container.js
import { ChangeDetectorRef, Component, HostBinding, ViewChild, ViewContainerRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../scroll-inertia/scroll_inertia.directive';

const _c0 = ["display_container"];
function DisplayContainerComponent_ng_template_0_Template(rf, ctx) { }
export class DisplayContainerComponent {
    constructor(cdr, _viewContainer) {
        this.cdr = cdr;
        this._viewContainer = _viewContainer;
        this.cssClass = 'igx-display-container';
        this.notVirtual = true;
    }
}
DisplayContainerComponent.ɵfac = function DisplayContainerComponent_Factory(t) { return new (t || DisplayContainerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
DisplayContainerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DisplayContainerComponent, selectors: [["igx-display-container"]], viewQuery: function DisplayContainerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true, ViewContainerRef);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._vcr = _t.first);
    } }, hostVars: 4, hostBindings: function DisplayContainerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.cssClass);
        ɵngcc0.ɵɵclassProp("igx-display-container--inactive", ctx.notVirtual);
    } }, decls: 2, vars: 2, consts: [["igxScrollInertia", "", 3, "IgxScrollInertiaScrollContainer", "IgxScrollInertiaDirection"], ["display_container", ""]], template: function DisplayContainerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, DisplayContainerComponent_ng_template_0_Template, 0, 0, "ng-template", 0, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("IgxScrollInertiaScrollContainer", ctx.scrollContainer)("IgxScrollInertiaDirection", ctx.scrollDirection);
    } }, directives: [ɵngcc1.IgxScrollInertiaDirective], encapsulation: 2 });
DisplayContainerComponent.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: ViewContainerRef }
];
DisplayContainerComponent.propDecorators = {
    _vcr: [{ type: ViewChild, args: ['display_container', { read: ViewContainerRef, static: true },] }],
    cssClass: [{ type: HostBinding, args: ['class',] }],
    notVirtual: [{ type: HostBinding, args: ['class.igx-display-container--inactive',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DisplayContainerComponent, [{
        type: Component,
        args: [{
                selector: 'igx-display-container',
                template: `
        <ng-template
            #display_container
            igxScrollInertia
            [IgxScrollInertiaScrollContainer]="scrollContainer"
            [IgxScrollInertiaDirection]="scrollDirection">
        </ng-template>
    `
            }]
    }], function () { return [{ type: ɵngcc0.ChangeDetectorRef }, { type: ɵngcc0.ViewContainerRef }]; }, { cssClass: [{
            type: HostBinding,
            args: ['class']
        }], notVirtual: [{
            type: HostBinding,
            args: ['class.igx-display-container--inactive']
        }], _vcr: [{
            type: ViewChild,
            args: ['display_container', { read: ViewContainerRef, static: true }]
        }] }); })();
