// igniteui-angular/esm2015/lib/directives/for-of/horizontal.virtual.helper.component.js
import { Component, ElementRef, HostBinding, Input, ViewChild, ViewContainerRef, ChangeDetectorRef, Inject, NgZone } from '@angular/core';
import { VirtualHelperBaseDirective } from './base.helper.component';
import { DOCUMENT } from '@angular/common';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = ["horizontal_container"];
export class HVirtualHelperComponent extends VirtualHelperBaseDirective {
    constructor(elementRef, cdr, _zone, document) {
        super(elementRef, cdr, _zone, document);
        this.elementRef = elementRef;
        this.cdr = cdr;
        this._zone = _zone;
        this.document = document;
        this.cssClasses = 'igx-vhelper--horizontal';
    }
    restoreScroll() {
        this.nativeElement.scrollLeft = this.scrollAmount;
    }
}
HVirtualHelperComponent.ɵfac = function HVirtualHelperComponent_Factory(t) { return new (t || HVirtualHelperComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.NgZone), ɵngcc0.ɵɵdirectiveInject(DOCUMENT)); };
HVirtualHelperComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: HVirtualHelperComponent, selectors: [["igx-horizontal-virtual-helper"]], viewQuery: function HVirtualHelperComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true, ViewContainerRef);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx._vcr = _t.first);
    } }, hostVars: 2, hostBindings: function HVirtualHelperComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.cssClasses);
    } }, inputs: { width: "width" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 2, vars: 2, consts: [[1, "igx-vhelper__placeholder-content"], ["horizontal_container", ""]], template: function HVirtualHelperComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0, 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("width", ctx.size, "px");
    } }, encapsulation: 2 });
HVirtualHelperComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef },
    { type: NgZone },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
HVirtualHelperComponent.propDecorators = {
    _vcr: [{ type: ViewChild, args: ['horizontal_container', { read: ViewContainerRef, static: true },] }],
    width: [{ type: Input }],
    cssClasses: [{ type: HostBinding, args: ['class',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HVirtualHelperComponent, [{
        type: Component,
        args: [{
                selector: 'igx-horizontal-virtual-helper',
                template: '<div #horizontal_container class="igx-vhelper__placeholder-content" [style.width.px]="size"></div>'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.ChangeDetectorRef }, { type: ɵngcc0.NgZone }, { type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }]; }, { cssClasses: [{
            type: HostBinding,
            args: ['class']
        }], _vcr: [{
            type: ViewChild,
            args: ['horizontal_container', { read: ViewContainerRef, static: true }]
        }], width: [{
            type: Input
        }] }); })();
