// igniteui-angular/esm2015/lib/directives/focus/focus.directive.js
import { Directive, ElementRef, Input, NgModule, Optional, Inject, Self } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as ɵngcc0 from '@angular/core';
export class IgxFocusDirective {
    constructor(element, comp) {
        this.element = element;
        this.comp = comp;
        this.focusState = true;
    }
    /**
     * Returns the state of the igxFocus.
     * ```typescript
     * @ViewChild('focusContainer', {read: IgxFocusDirective})
     * public igxFocus: IgxFocusDirective;
     * let isFocusOn = this.igxFocus.focused;
     * ```
     * @memberof IgxFocusDirective
     */
    get focused() {
        return this.focusState;
    }
    /**
     * Sets the state of the igxFocus.
     * ```html
     * <igx-input-group >
     *  <input #focusContainer igxInput [igxFocus]="true"/>
     * </igx-input-group>
     * ```
     * @memberof IgxFocusDirective
     */
    set focused(val) {
        this.focusState = val;
        this.trigger();
    }
    /**
     * Gets the native element of the igxFocus.
     * ```typescript
     * @ViewChild('focusContainer', {read: IgxFocusDirective})
     * public igxFocus: IgxFocusDirective;
     * let igxFocusNativeElement = this.igxFocus.nativeElement;
     * ```
     * @memberof IgxFocusDirective
     */
    get nativeElement() {
        if (this.comp && this.comp[0] && this.comp[0].getEditElement) {
            return this.comp[0].getEditElement();
        }
        return this.element.nativeElement;
    }
    /**
     * Triggers the igxFocus state.
     * ```typescript
     * @ViewChild('focusContainer', {read: IgxFocusDirective})
     * public igxFocus: IgxFocusDirective;
     * this.igxFocus.trigger();
     * ```
     * @memberof IgxFocusDirective
     */
    trigger() {
        if (this.focusState) {
            requestAnimationFrame(() => this.nativeElement.focus({ preventScroll: true }));
        }
    }
}
IgxFocusDirective.ɵfac = function IgxFocusDirective_Factory(t) { return new (t || IgxFocusDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(NG_VALUE_ACCESSOR, 10)); };
IgxFocusDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxFocusDirective, selectors: [["", "igxFocus", ""]], inputs: { focused: ["igxFocus", "focused"] }, exportAs: ["igxFocus"] });
IgxFocusDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Array, decorators: [{ type: Inject, args: [NG_VALUE_ACCESSOR,] }, { type: Self }, { type: Optional }] }
];
IgxFocusDirective.propDecorators = {
    focused: [{ type: Input, args: ['igxFocus',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxFocusDirective, [{
        type: Directive,
        args: [{
                exportAs: 'igxFocus',
                selector: '[igxFocus]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: Array, decorators: [{
                type: Inject,
                args: [NG_VALUE_ACCESSOR]
            }, {
                type: Self
            }, {
                type: Optional
            }] }]; }, { focused: [{
            type: Input,
            args: ['igxFocus']
        }] }); })();
/**
 * @hidden
 */
export class IgxFocusModule {
}
IgxFocusModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxFocusModule });
IgxFocusModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxFocusModule_Factory(t) { return new (t || IgxFocusModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxFocusModule, { declarations: [IgxFocusDirective], exports: [IgxFocusDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxFocusModule, [{
        type: NgModule,
        args: [{
                declarations: [IgxFocusDirective],
                exports: [IgxFocusDirective]
            }]
    }], null, null); })();
