// igniteui-angular/esm2015/lib/grids/column-chooser-item-base.js
import { EventEmitter, Input, Output, Directive } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class ItemPropertyValueChangedDirective {
    constructor(propName) {
        this.valueChanged = new EventEmitter();
        this._propName = propName;
    }
    get object() {
        return this._object;
    }
    set object(value) {
        if (value) {
            this._object = value;
        }
    }
    get value() {
        return (this.object) ? this.object[this._propName] : null;
    }
    set value(value) {
        this.onValueChanged(value);
    }
    onValueChanged(value) {
        const currentValue = this.value;
        if (value !== currentValue) {
            this.object[this._propName] = value;
            this.valueChanged.emit({ oldValue: currentValue, newValue: value });
        }
    }
}
ItemPropertyValueChangedDirective.ɵfac = function ItemPropertyValueChangedDirective_Factory(t) { return new (t || ItemPropertyValueChangedDirective)(ɵngcc0.ɵɵdirectiveInject(String)); };
ItemPropertyValueChangedDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ItemPropertyValueChangedDirective, inputs: { value: "value" }, outputs: { valueChanged: "valueChanged" } });
ItemPropertyValueChangedDirective.ctorParameters = () => [
    { type: String }
];
ItemPropertyValueChangedDirective.propDecorators = {
    value: [{ type: Input }],
    valueChanged: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ItemPropertyValueChangedDirective, [{
        type: Directive
    }], function () { return [{ type: String }]; }, { valueChanged: [{
            type: Output
        }], value: [{
            type: Input
        }] }); })();
/** @hidden */
export class ColumnChooserItemBaseDirective extends ItemPropertyValueChangedDirective {
    constructor(prop) {
        super(prop);
        this.prop = prop;
        this.indentation = 30;
    }
    get column() {
        return this.object;
    }
    set column(value) {
        if (value) {
            this.object = value;
        }
    }
    get name() {
        return (this.column) ? ((this.column.header) ? this.column.header : this.column.field) : '';
    }
    get level() {
        return this.column.level;
    }
    get calcIndent() {
        return this.indentation * this.level;
    }
}
ColumnChooserItemBaseDirective.ɵfac = function ColumnChooserItemBaseDirective_Factory(t) { return new (t || ColumnChooserItemBaseDirective)(ɵngcc0.ɵɵdirectiveInject(String)); };
ColumnChooserItemBaseDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ColumnChooserItemBaseDirective, inputs: { indentation: "indentation", column: "column", container: "container" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
ColumnChooserItemBaseDirective.ctorParameters = () => [
    { type: String }
];
ColumnChooserItemBaseDirective.propDecorators = {
    column: [{ type: Input }],
    indentation: [{ type: Input }],
    container: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ColumnChooserItemBaseDirective, [{
        type: Directive
    }], function () { return [{ type: String }]; }, { indentation: [{
            type: Input
        }], column: [{
            type: Input
        }], container: [{
            type: Input
        }] }); })();
