// igniteui-angular/esm2015/lib/grids/hierarchical-grid/hierarchical-grid.module.js
import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { IgxGridModule } from '../grid/grid.module';
import { IgxHierarchicalGridComponent } from './hierarchical-grid.component';
import { IgxHierarchicalRowComponent } from './hierarchical-row.component';
import { IgxGridHierarchicalPipe, IgxGridHierarchicalPagingPipe } from './hierarchical-grid.pipes';
import { IgxRowIslandComponent } from './row-island.component';
import { IgxChildGridRowComponent } from './child-grid-row.component';
import { IgxHierarchicalGridCellComponent } from './hierarchical-cell.component';
import { IgxGridComponent } from '../grid/grid.component';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../grid/grid.component';
import * as ɵngcc2 from '../grid/expandable-cell.component';
import * as ɵngcc3 from '../grid/groupby-row.component';
import * as ɵngcc4 from '../grid/grid-row.component';
import * as ɵngcc5 from '../grid/grid.directives';
import * as ɵngcc6 from '../cell.component';
import * as ɵngcc7 from '../grid.rowEdit.directive';
import * as ɵngcc8 from '../grid.common';
import * as ɵngcc9 from '../hiding/column-hiding.component';
import * as ɵngcc10 from '../pinning/column-pinning.component';
import * as ɵngcc11 from '../column-actions/column-hiding.directive';
import * as ɵngcc12 from '../column-actions/column-pinning.directive';
import * as ɵngcc13 from '../column-actions/column-actions.component';
import * as ɵngcc14 from '../columns/templates.directive';
import * as ɵngcc15 from '../columns/column.component';
import * as ɵngcc16 from '../columns/column-group.component';
import * as ɵngcc17 from '../columns/column-layout.component';
import * as ɵngcc18 from '../headers/grid-header.component';
import * as ɵngcc19 from '../headers/grid-header-group.component';
import * as ɵngcc20 from '../filtering/base/grid-filtering-cell.component';
import * as ɵngcc21 from '../filtering/base/grid-filtering-row.component';
import * as ɵngcc22 from '../filtering/excel-style/grid.excel-style-filtering.component';
import * as ɵngcc23 from '../filtering/excel-style/excel-style-search.component';
import * as ɵngcc24 from '../filtering/excel-style/excel-style-date-expression.component';
import * as ɵngcc25 from '../filtering/excel-style/excel-style-header.component';
import * as ɵngcc26 from '../filtering/excel-style/excel-style-sorting.component';
import * as ɵngcc27 from '../filtering/excel-style/excel-style-pinning.component';
import * as ɵngcc28 from '../filtering/excel-style/excel-style-hiding.component';
import * as ɵngcc29 from '../filtering/excel-style/excel-style-selecting.component';
import * as ɵngcc30 from '../filtering/excel-style/excel-style-clear-filters.component';
import * as ɵngcc31 from '../filtering/excel-style/excel-style-conditional-filter.component';
import * as ɵngcc32 from '../filtering/excel-style/excel-style-moving.component';
import * as ɵngcc33 from '../row-drag.directive';
import * as ɵngcc34 from '../../paginator/paginator.component';
import * as ɵngcc35 from '../grid-footer/grid-footer.component';
import * as ɵngcc36 from '../resizing/resizer.component';
import * as ɵngcc37 from '../resizing/resize-handle.directive';
import * as ɵngcc38 from '../resizing/resizer.directive';
import * as ɵngcc39 from '../moving/moving.drop.directive';
import * as ɵngcc40 from '../moving/moving.drag.directive';
import * as ɵngcc41 from '../selection/row-selectors';
import * as ɵngcc42 from '../selection/drag-select.directive';
import * as ɵngcc43 from '../summaries/summary-cell.component';
import * as ɵngcc44 from '../summaries/summary-row.component';
import * as ɵngcc45 from '../toolbar/grid-toolbar.component';
import * as ɵngcc46 from '../toolbar/toolbar.directive';
import * as ɵngcc47 from '../filtering/advanced-filtering/advanced-filtering-dialog.component';
import * as ɵngcc48 from '@angular/common';
import * as ɵngcc49 from '@angular/forms';
import * as ɵngcc50 from '../../directives/button/button.directive';
import * as ɵngcc51 from '../../date-picker/date-picker.component';
import * as ɵngcc52 from '../../date-picker/date-picker.directives';
import * as ɵngcc53 from '../../icon/icon.component';
import * as ɵngcc54 from '../../directives/ripple/ripple.directive';
import * as ɵngcc55 from '../../input-group/input-group.component';
import * as ɵngcc56 from '../../directives/hint/hint.directive';
import * as ɵngcc57 from '../../directives/input/input.directive';
import * as ɵngcc58 from '../../directives/label/label.directive';
import * as ɵngcc59 from '../../directives/prefix/prefix.directive';
import * as ɵngcc60 from '../../directives/suffix/suffix.directive';
import * as ɵngcc61 from '../../directives/focus/focus.directive';
import * as ɵngcc62 from '../../directives/toggle/toggle.directive';
import * as ɵngcc63 from '../../directives/for-of/for_of.directive';
import * as ɵngcc64 from '../../directives/template-outlet/template_outlet.directive';
import * as ɵngcc65 from '../../directives/text-highlight/text-highlight.directive';
import * as ɵngcc66 from '../state.directive';
import * as ɵngcc67 from '../../directives/text-selection/text-selection.directive';
import * as ɵngcc68 from '../../checkbox/checkbox.component';
import * as ɵngcc69 from '../../badge/badge.component';
import * as ɵngcc70 from '../../chips/chips-area.component';
import * as ɵngcc71 from '../../chips/chip.component';
import * as ɵngcc72 from '../../directives/drag-drop/drag-drop.directive';
import * as ɵngcc73 from '../../drop-down/drop-down.component';
import * as ɵngcc74 from '../../drop-down/drop-down-item.component';
import * as ɵngcc75 from '../../drop-down/drop-down-group.component';
import * as ɵngcc76 from '../../drop-down/drop-down-navigation.directive';
import * as ɵngcc77 from '../../buttonGroup/buttonGroup.component';
import * as ɵngcc78 from '../../progressbar/progressbar.component';
import * as ɵngcc79 from '../../progressbar/progressbar.common';
import * as ɵngcc80 from '../../select/select.component';
import * as ɵngcc81 from '../../select/select-group.component';
import * as ɵngcc82 from '../../select/select-item.component';
import * as ɵngcc83 from '../../select/select-navigation.directive';
import * as ɵngcc84 from '../../snackbar/snackbar.component';
import * as ɵngcc85 from '../grid/grid.pipes';
import * as ɵngcc86 from '../grid/grid.summary.pipe';
import * as ɵngcc87 from '../grid/grid.details.pipe';
import * as ɵngcc88 from '../common/pipes';
import * as ɵngcc89 from '../summaries/grid-root-summary.pipe';
import * as ɵngcc90 from '../../date-picker/date-picker.pipes';
export class IgxHierarchicalGridModule {
}
IgxHierarchicalGridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxHierarchicalGridModule });
IgxHierarchicalGridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxHierarchicalGridModule_Factory(t) { return new (t || IgxHierarchicalGridModule)(); }, imports: [[
            IgxGridModule,
        ], IgxGridModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxHierarchicalGridModule, { declarations: function () { return [IgxHierarchicalGridComponent,
        IgxHierarchicalRowComponent,
        IgxRowIslandComponent,
        IgxChildGridRowComponent,
        IgxHierarchicalGridCellComponent,
        IgxGridHierarchicalPipe,
        IgxGridHierarchicalPagingPipe]; }, imports: function () { return [IgxGridModule]; }, exports: function () { return [IgxGridModule,
        IgxHierarchicalGridComponent,
        IgxHierarchicalRowComponent,
        IgxHierarchicalGridCellComponent,
        IgxRowIslandComponent,
        IgxChildGridRowComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxHierarchicalGridModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxHierarchicalGridComponent,
                    IgxHierarchicalRowComponent,
                    IgxRowIslandComponent,
                    IgxChildGridRowComponent,
                    IgxHierarchicalGridCellComponent,
                    IgxGridHierarchicalPipe,
                    IgxGridHierarchicalPagingPipe
                ],
                exports: [
                    IgxGridModule,
                    IgxHierarchicalGridComponent,
                    IgxHierarchicalRowComponent,
                    IgxHierarchicalGridCellComponent,
                    IgxRowIslandComponent,
                    IgxChildGridRowComponent
                ],
                imports: [
                    IgxGridModule,
                ],
                entryComponents: [
                    IgxGridComponent
                ],
                schemas: [CUSTOM_ELEMENTS_SCHEMA]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(IgxChildGridRowComponent, [ɵngcc1.IgxGridComponent, ɵngcc2.IgxGridExpandableCellComponent, ɵngcc3.IgxGridGroupByRowComponent, ɵngcc4.IgxGridRowComponent, ɵngcc5.IgxGroupByRowTemplateDirective, ɵngcc5.IgxGridDetailTemplateDirective, ɵngcc5.IgxRowExpandedIndicatorDirective, ɵngcc5.IgxRowCollapsedIndicatorDirective, ɵngcc5.IgxHeaderExpandIndicatorDirective, ɵngcc5.IgxHeaderCollapseIndicatorDirective, ɵngcc5.IgxGroupAreaDropDirective, ɵngcc6.IgxGridCellComponent, ɵngcc7.IgxRowEditTemplateDirective, ɵngcc7.IgxRowEditActionsDirective, ɵngcc7.IgxRowEditTextDirective, ɵngcc7.IgxRowEditTabStopDirective, ɵngcc8.IgxGridBodyDirective, ɵngcc9.IgxColumnHidingComponent, ɵngcc10.IgxColumnPinningComponent, ɵngcc11.IgxColumnHidingDirective, ɵngcc12.IgxColumnPinningDirective, ɵngcc13.IgxColumnActionsComponent, ɵngcc14.IgxFilterCellTemplateDirective, ɵngcc14.IgxCellTemplateDirective, ɵngcc14.IgxCellHeaderTemplateDirective, ɵngcc14.IgxCellFooterTemplateDirective, ɵngcc14.IgxCellEditorTemplateDirective, ɵngcc14.IgxCollapsibleIndicatorTemplateDirective, ɵngcc15.IgxColumnComponent, ɵngcc16.IgxColumnGroupComponent, ɵngcc17.IgxColumnLayoutComponent, ɵngcc18.IgxGridHeaderComponent, ɵngcc19.IgxGridHeaderGroupComponent, ɵngcc20.IgxGridFilteringCellComponent, ɵngcc21.IgxGridFilteringRowComponent, ɵngcc22.IgxGridExcelStyleFilteringComponent, ɵngcc22.IgxExcelStyleColumnOperationsTemplateDirective, ɵngcc22.IgxExcelStyleFilterOperationsTemplateDirective, ɵngcc23.IgxExcelStyleLoadingValuesTemplateDirective, ɵngcc24.IgxExcelStyleDateExpressionComponent, ɵngcc25.IgxExcelStyleHeaderComponent, ɵngcc26.IgxExcelStyleSortingComponent, ɵngcc27.IgxExcelStylePinningComponent, ɵngcc28.IgxExcelStyleHidingComponent, ɵngcc29.IgxExcelStyleSelectingComponent, ɵngcc30.IgxExcelStyleClearFiltersComponent, ɵngcc31.IgxExcelStyleConditionalFilterComponent, ɵngcc32.IgxExcelStyleMovingComponent, ɵngcc23.IgxExcelStyleSearchComponent, ɵngcc33.IgxRowDragDirective, ɵngcc33.IgxDragIndicatorIconDirective, ɵngcc33.IgxRowDragGhostDirective, ɵngcc34.IgxPaginatorComponent, ɵngcc35.IgxGridFooterComponent, ɵngcc36.IgxGridColumnResizerComponent, ɵngcc37.IgxResizeHandleDirective, ɵngcc38.IgxColumnResizerDirective, ɵngcc39.IgxColumnMovingDropDirective, ɵngcc40.IgxColumnMovingDragDirective, ɵngcc41.IgxRowSelectorDirective, ɵngcc41.IgxHeadSelectorDirective, ɵngcc42.IgxGridDragSelectDirective, ɵngcc43.IgxSummaryCellComponent, ɵngcc44.IgxSummaryRowComponent, ɵngcc45.IgxGridToolbarComponent, ɵngcc46.IgxGridToolbarCustomContentDirective, ɵngcc47.IgxAdvancedFilteringDialogComponent, ɵngcc48.NgClass, ɵngcc48.NgComponentOutlet, ɵngcc48.NgForOf, ɵngcc48.NgIf, ɵngcc48.NgTemplateOutlet, ɵngcc48.NgStyle, ɵngcc48.NgSwitch, ɵngcc48.NgSwitchCase, ɵngcc48.NgSwitchDefault, ɵngcc48.NgPlural, ɵngcc48.NgPluralCase, ɵngcc49.ɵangular_packages_forms_forms_y, ɵngcc49.NgSelectOption, ɵngcc49.ɵangular_packages_forms_forms_x, ɵngcc49.DefaultValueAccessor, ɵngcc49.NumberValueAccessor, ɵngcc49.RangeValueAccessor, ɵngcc49.CheckboxControlValueAccessor, ɵngcc49.SelectControlValueAccessor, ɵngcc49.SelectMultipleControlValueAccessor, ɵngcc49.RadioControlValueAccessor, ɵngcc49.NgControlStatus, ɵngcc49.NgControlStatusGroup, ɵngcc49.RequiredValidator, ɵngcc49.MinLengthValidator, ɵngcc49.MaxLengthValidator, ɵngcc49.PatternValidator, ɵngcc49.CheckboxRequiredValidator, ɵngcc49.EmailValidator, ɵngcc49.NgModel, ɵngcc49.NgModelGroup, ɵngcc49.NgForm, ɵngcc50.IgxButtonDirective, ɵngcc51.IgxDatePickerComponent, ɵngcc52.IgxDatePickerTemplateDirective, ɵngcc52.IgxDatePickerActionsDirective, ɵngcc53.IgxIconComponent, ɵngcc54.IgxRippleDirective, ɵngcc55.IgxInputGroupComponent, ɵngcc56.IgxHintDirective, ɵngcc57.IgxInputDirective, ɵngcc58.IgxLabelDirective, ɵngcc59.IgxPrefixDirective, ɵngcc60.IgxSuffixDirective, ɵngcc61.IgxFocusDirective, ɵngcc62.IgxToggleDirective, ɵngcc62.IgxToggleActionDirective, ɵngcc62.IgxOverlayOutletDirective, ɵngcc63.IgxForOfDirective, ɵngcc63.IgxGridForOfDirective, ɵngcc64.IgxTemplateOutletDirective, ɵngcc65.IgxTextHighlightDirective, ɵngcc66.IgxGridStateDirective, ɵngcc67.IgxTextSelectionDirective, ɵngcc68.IgxCheckboxComponent, ɵngcc68.IgxCheckboxRequiredDirective, ɵngcc69.IgxBadgeComponent, ɵngcc70.IgxChipsAreaComponent, ɵngcc71.IgxChipComponent, ɵngcc72.IgxDragDirective, ɵngcc72.IgxDropDirective, ɵngcc72.IgxDragHandleDirective, ɵngcc72.IgxDragIgnoreDirective, ɵngcc73.IgxDropDownComponent, ɵngcc74.IgxDropDownItemComponent, ɵngcc75.IgxDropDownGroupComponent, ɵngcc76.IgxDropDownItemNavigationDirective, ɵngcc77.IgxButtonGroupComponent, ɵngcc78.IgxLinearProgressBarComponent, ɵngcc78.IgxCircularProgressBarComponent, ɵngcc79.IgxProcessBarTextTemplateDirective, ɵngcc79.IgxProgressBarGradientDirective, ɵngcc80.IgxSelectComponent, ɵngcc80.IgxSelectFooterDirective, ɵngcc81.IgxSelectGroupComponent, ɵngcc80.IgxSelectHeaderDirective, ɵngcc82.IgxSelectItemComponent, ɵngcc83.IgxSelectItemNavigationDirective, ɵngcc80.IgxSelectToggleIconDirective, ɵngcc84.IgxSnackbarComponent, IgxHierarchicalGridComponent,
    IgxHierarchicalRowComponent,
    IgxRowIslandComponent,
    IgxChildGridRowComponent,
    IgxHierarchicalGridCellComponent], [ɵngcc85.IgxGridGroupingPipe, ɵngcc85.IgxGridPagingPipe, ɵngcc85.IgxGridSortingPipe, ɵngcc85.IgxGridFilteringPipe, ɵngcc86.IgxGridSummaryPipe, ɵngcc87.IgxGridDetailsPipe, ɵngcc88.IgxGridFilterConditionPipe, ɵngcc88.IgxGridTransactionPipe, ɵngcc88.IgxGridNotGroupedPipe, ɵngcc88.IgxGridTopLevelColumns, ɵngcc88.IgxGridCellStylesPipe, ɵngcc88.IgxGridCellStyleClassesPipe, ɵngcc88.IgxGridPaginatorOptionsPipe, ɵngcc88.IgxHasVisibleColumnsPipe, ɵngcc88.IgxGridRowPinningPipe, ɵngcc88.IgxColumnActionEnabledPipe, ɵngcc88.IgxFilterActionColumnsPipe, ɵngcc88.IgxSortActionColumnsPipe, ɵngcc88.IgxGridDataMapperPipe, ɵngcc88.IgxStringReplacePipe, ɵngcc88.IgxGridTransactionStatePipe, ɵngcc88.IgxGridAddRowPipe, ɵngcc88.IgxColumnFormatterPipe, ɵngcc89.IgxSummaryDataPipe, ɵngcc48.AsyncPipe, ɵngcc48.UpperCasePipe, ɵngcc48.LowerCasePipe, ɵngcc48.JsonPipe, ɵngcc48.SlicePipe, ɵngcc48.DecimalPipe, ɵngcc48.PercentPipe, ɵngcc48.TitleCasePipe, ɵngcc48.CurrencyPipe, ɵngcc48.DatePipe, ɵngcc48.I18nPluralPipe, ɵngcc48.I18nSelectPipe, ɵngcc48.KeyValuePipe, ɵngcc90.DatePickerDisplayValuePipe, ɵngcc90.DatePickerInputValuePipe, IgxGridHierarchicalPipe,
    IgxGridHierarchicalPagingPipe]);
