// igniteui-angular/esm2015/lib/combo/combo.directives.js
import { Directive } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxComboHeaderDirective {
}
IgxComboHeaderDirective.ɵfac = function IgxComboHeaderDirective_Factory(t) { return new (t || IgxComboHeaderDirective)(); };
IgxComboHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboHeaderDirective, selectors: [["", "igxComboHeader", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboHeaderDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboHeader]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboFooterDirective {
}
IgxComboFooterDirective.ɵfac = function IgxComboFooterDirective_Factory(t) { return new (t || IgxComboFooterDirective)(); };
IgxComboFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboFooterDirective, selectors: [["", "igxComboFooter", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboFooterDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboFooter]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboItemDirective {
}
IgxComboItemDirective.ɵfac = function IgxComboItemDirective_Factory(t) { return new (t || IgxComboItemDirective)(); };
IgxComboItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboItemDirective, selectors: [["", "igxComboItem", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboItemDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboItem]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboEmptyDirective {
}
IgxComboEmptyDirective.ɵfac = function IgxComboEmptyDirective_Factory(t) { return new (t || IgxComboEmptyDirective)(); };
IgxComboEmptyDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboEmptyDirective, selectors: [["", "igxComboEmpty", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboEmptyDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboEmpty]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboHeaderItemDirective {
}
IgxComboHeaderItemDirective.ɵfac = function IgxComboHeaderItemDirective_Factory(t) { return new (t || IgxComboHeaderItemDirective)(); };
IgxComboHeaderItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboHeaderItemDirective, selectors: [["", "igxComboHeaderItem", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboHeaderItemDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboHeaderItem]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboAddItemDirective {
}
IgxComboAddItemDirective.ɵfac = function IgxComboAddItemDirective_Factory(t) { return new (t || IgxComboAddItemDirective)(); };
IgxComboAddItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboAddItemDirective, selectors: [["", "igxComboAddItem", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboAddItemDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboAddItem]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboToggleIconDirective {
}
IgxComboToggleIconDirective.ɵfac = function IgxComboToggleIconDirective_Factory(t) { return new (t || IgxComboToggleIconDirective)(); };
IgxComboToggleIconDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboToggleIconDirective, selectors: [["", "igxComboToggleIcon", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboToggleIconDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboToggleIcon]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxComboClearIconDirective {
}
IgxComboClearIconDirective.ɵfac = function IgxComboClearIconDirective_Factory(t) { return new (t || IgxComboClearIconDirective)(); };
IgxComboClearIconDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxComboClearIconDirective, selectors: [["", "igxComboClearIcon", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboClearIconDirective, [{
        type: Directive,
        args: [{
                selector: '[igxComboClearIcon]'
            }]
    }], null, null); })();
