// igniteui-angular/esm2015/lib/combo/combo.api.js
import { Injectable } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxComboAPIService {
    constructor() {
        this.disableTransitions = false;
    }
    register(combo) {
        this.combo = combo;
    }
    clear() {
        this.combo = null;
    }
    get valueKey() {
        return this.combo.valueKey !== null && this.combo.valueKey !== undefined ? this.combo.valueKey : null;
    }
    get item_focusable() {
        return false;
    }
    get isRemote() {
        return this.combo.isRemote;
    }
    add_custom_item() {
        if (!this.combo) {
            return;
        }
        this.combo.addItemToCollection();
    }
    get comboID() {
        return this.combo.id;
    }
    set_selected_item(itemID, event) {
        const selected = this.combo.isItemSelected(itemID);
        if (itemID === null || itemID === undefined) {
            return;
        }
        if (!selected) {
            this.combo.selectItems([itemID], false, event);
        }
        else {
            this.combo.deselectItems([itemID], event);
        }
    }
    is_item_selected(itemID) {
        return this.combo.isItemSelected(itemID);
    }
}
IgxComboAPIService.ɵfac = function IgxComboAPIService_Factory(t) { return new (t || IgxComboAPIService)(); };
IgxComboAPIService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: IgxComboAPIService, factory: IgxComboAPIService.ɵfac });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboAPIService, [{
        type: Injectable
    }], function () { return []; }, null); })();
