// igniteui-angular/esm2015/lib/combo/combo-add-item.component.js
import { IgxComboItemComponent } from './combo-item.component';
import { Component } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class IgxComboAddItemComponent extends IgxComboItemComponent {
    get selected() {
        return false;
    }
    set selected(value) {
    }
    /**
     * @inheritdoc
     */
    clicked(event) {
        this.comboAPI.disableTransitions = false;
        this.comboAPI.add_custom_item();
    }
}
IgxComboAddItemComponent.ɵfac = function IgxComboAddItemComponent_Factory(t) { return ɵIgxComboAddItemComponent_BaseFactory(t || IgxComboAddItemComponent); };
IgxComboAddItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxComboAddItemComponent, selectors: [["igx-combo-add-item"]], features: [ɵngcc0.ɵɵProvidersFeature([{ provide: IgxComboItemComponent, useExisting: IgxComboAddItemComponent }]), ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function IgxComboAddItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
const ɵIgxComboAddItemComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(IgxComboAddItemComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxComboAddItemComponent, [{
        type: Component,
        args: [{
                selector: 'igx-combo-add-item',
                template: '<ng-content></ng-content>',
                providers: [{ provide: IgxComboItemComponent, useExisting: IgxComboAddItemComponent }]
            }]
    }], null, null); })();
