// igniteui-angular/esm2015/lib/progressbar/progressbar.common.js
import { Directive, TemplateRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class IgxProcessBarTextTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxProcessBarTextTemplateDirective.ɵfac = function IgxProcessBarTextTemplateDirective_Factory(t) { return new (t || IgxProcessBarTextTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxProcessBarTextTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxProcessBarTextTemplateDirective, selectors: [["", "igxProcessBarText", ""]] });
IgxProcessBarTextTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxProcessBarTextTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxProcessBarText]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
export class IgxProgressBarGradientDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxProgressBarGradientDirective.ɵfac = function IgxProgressBarGradientDirective_Factory(t) { return new (t || IgxProgressBarGradientDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxProgressBarGradientDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxProgressBarGradientDirective, selectors: [["", "igxProgressBarGradient", ""]] });
IgxProgressBarGradientDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxProgressBarGradientDirective, [{
        type: Directive,
        args: [{
                selector: '[igxProgressBarGradient]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
