// igniteui-angular/esm2015/lib/directives/prefix/prefix.directive.js
import { NgModule, Directive } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxPrefixDirective {
}
IgxPrefixDirective.ɵfac = function IgxPrefixDirective_Factory(t) { return new (t || IgxPrefixDirective)(); };
IgxPrefixDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxPrefixDirective, selectors: [["igx-prefix"], ["", "igxPrefix", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxPrefixDirective, [{
        type: Directive,
        args: [{
                selector: 'igx-prefix,[igxPrefix]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxPrefixModule {
}
IgxPrefixModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxPrefixModule });
IgxPrefixModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxPrefixModule_Factory(t) { return new (t || IgxPrefixModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxPrefixModule, { declarations: [IgxPrefixDirective], exports: [IgxPrefixDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxPrefixModule, [{
        type: NgModule,
        args: [{
                declarations: [IgxPrefixDirective],
                exports: [IgxPrefixDirective]
            }]
    }], null, null); })();
