// igniteui-angular/esm2015/lib/date-picker/calendar-container.component.js
import { Component, ViewChild, Input, Output, EventEmitter, HostListener, HostBinding } from '@angular/core';
import { InteractionMode } from '../core/enums';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../calendar/calendar.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../directives/button/button.directive';
import * as ɵngcc4 from '../directives/ripple/ripple.directive';

const _c0 = ["calendar"];
function IgxCalendarContainerComponent_ng_template_0_div_0_button_1_Template(rf, ctx) { if (rf & 1) {
    const _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 6, 7);
    ɵngcc0.ɵɵlistener("click", function IgxCalendarContainerComponent_ng_template_0_div_0_button_1_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r9); const ctx_r8 = ɵngcc0.ɵɵnextContext(3); return ctx_r8.closeCalendar(); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r5.cancelButtonLabel, " ");
} }
function IgxCalendarContainerComponent_ng_template_0_div_0_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 6, 8);
    ɵngcc0.ɵɵlistener("click", function IgxCalendarContainerComponent_ng_template_0_div_0_button_2_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(3); return ctx_r11.triggerTodaySelection(); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r6.todayButtonLabel, " ");
} }
function IgxCalendarContainerComponent_ng_template_0_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵtemplate(1, IgxCalendarContainerComponent_ng_template_0_div_0_button_1_Template, 3, 1, "button", 5);
    ɵngcc0.ɵɵtemplate(2, IgxCalendarContainerComponent_ng_template_0_div_0_button_2_Template, 3, 1, "button", 5);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.cancelButtonLabel);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r4.todayButtonLabel);
} }
function IgxCalendarContainerComponent_ng_template_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, IgxCalendarContainerComponent_ng_template_0_div_0_Template, 3, 2, "div", 3);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.cancelButtonLabel || ctx_r1.todayButtonLabel);
} }
function IgxCalendarContainerComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
export class IgxCalendarContainerComponent {
    constructor() {
        this.mode = InteractionMode.Dialog;
        this.vertical = false;
        this.onClose = new EventEmitter();
        this.onTodaySelection = new EventEmitter();
        this.styleClass = 'igx-date-picker';
    }
    get dropdownCSS() {
        return this.mode === InteractionMode.DropDown;
    }
    get verticalCSS() {
        return this.vertical && this.mode === InteractionMode.Dialog;
    }
    onEscape(event) {
        event.preventDefault();
        this.onClose.emit();
    }
    /**
     * Returns whether the date-picker is in readonly dialog mode.
     *
     * @hidden
     */
    get isReadonly() {
        return this.mode === InteractionMode.Dialog;
    }
    /**
     * Emits close event for the calendar.
     */
    closeCalendar() {
        this.onClose.emit();
    }
    /**
     * Emits today selection event for the calendar.
     */
    triggerTodaySelection() {
        this.onTodaySelection.emit();
    }
}
IgxCalendarContainerComponent.ɵfac = function IgxCalendarContainerComponent_Factory(t) { return new (t || IgxCalendarContainerComponent)(); };
IgxCalendarContainerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxCalendarContainerComponent, selectors: [["igx-calendar-container"]], viewQuery: function IgxCalendarContainerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(_c0, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.calendar = _t.first);
    } }, hostVars: 6, hostBindings: function IgxCalendarContainerComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keydown.esc", function IgxCalendarContainerComponent_keydown_esc_HostBindingHandler($event) { return ctx.onEscape($event); })("keydown.alt.arrowup", function IgxCalendarContainerComponent_keydown_alt_arrowup_HostBindingHandler($event) { return ctx.onEscape($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-date-picker", ctx.styleClass)("igx-date-picker--dropdown", ctx.dropdownCSS)("igx-date-picker--vertical", ctx.verticalCSS);
    } }, inputs: { mode: "mode", vertical: "vertical", cancelButtonLabel: "cancelButtonLabel", todayButtonLabel: "todayButtonLabel", datePickerActions: "datePickerActions" }, outputs: { onClose: "onClose", onTodaySelection: "onTodaySelection" }, decls: 5, vars: 1, consts: [["defaultDatePickerActions", ""], ["calendar", ""], [4, "ngTemplateOutlet"], ["class", "igx-date-picker__buttons", 4, "ngIf"], [1, "igx-date-picker__buttons"], ["igxButton", "flat", "igxRipple", "", 3, "click", 4, "ngIf"], ["igxButton", "flat", "igxRipple", "", 3, "click"], ["closeButton", ""], ["todayButton", ""]], template: function IgxCalendarContainerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxCalendarContainerComponent_ng_template_0_Template, 1, 1, "ng-template", null, 0, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵelement(2, "igx-calendar", null, 1);
        ɵngcc0.ɵɵtemplate(4, IgxCalendarContainerComponent_ng_container_4_Template, 1, 0, "ng-container", 2);
    } if (rf & 2) {
        const _r0 = ɵngcc0.ɵɵreference(1);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx.datePickerActions ? ctx.datePickerActions.template : _r0);
    } }, directives: [ɵngcc1.IgxCalendarComponent, ɵngcc2.NgTemplateOutlet, ɵngcc2.NgIf, ɵngcc3.IgxButtonDirective, ɵngcc4.IgxRippleDirective], styles: ["[_nghost-%COMP%] {display: block;}"] });
IgxCalendarContainerComponent.propDecorators = {
    calendar: [{ type: ViewChild, args: ['calendar', { static: true },] }],
    mode: [{ type: Input }],
    vertical: [{ type: Input }],
    cancelButtonLabel: [{ type: Input }],
    todayButtonLabel: [{ type: Input }],
    datePickerActions: [{ type: Input }],
    onClose: [{ type: Output }],
    onTodaySelection: [{ type: Output }],
    styleClass: [{ type: HostBinding, args: ['class.igx-date-picker',] }],
    dropdownCSS: [{ type: HostBinding, args: ['class.igx-date-picker--dropdown',] }],
    verticalCSS: [{ type: HostBinding, args: ['class.igx-date-picker--vertical',] }],
    onEscape: [{ type: HostListener, args: ['keydown.esc', ['$event'],] }, { type: HostListener, args: ['keydown.alt.arrowup', ['$event'],] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxCalendarContainerComponent, [{
        type: Component,
        args: [{
                selector: 'igx-calendar-container',
                template: "<ng-template #defaultDatePickerActions>\n    <div *ngIf=\"cancelButtonLabel || todayButtonLabel\" class=\"igx-date-picker__buttons\">\n        <button #closeButton *ngIf=\"cancelButtonLabel\" igxButton=\"flat\" igxRipple (click)=\"closeCalendar()\">\n            {{ cancelButtonLabel }}\n        </button>\n        <button #todayButton *ngIf=\"todayButtonLabel\" igxButton=\"flat\" igxRipple (click)=\"triggerTodaySelection()\">\n            {{ todayButtonLabel }}\n        </button>\n    </div>\n</ng-template>\n<igx-calendar #calendar></igx-calendar>\n<ng-container *ngTemplateOutlet=\"datePickerActions ? datePickerActions.template : defaultDatePickerActions\"></ng-container>\n",
                styles: [':host {display: block;}']
            }]
    }], function () { return []; }, { mode: [{
            type: Input
        }], vertical: [{
            type: Input
        }], onClose: [{
            type: Output
        }], onTodaySelection: [{
            type: Output
        }], styleClass: [{
            type: HostBinding,
            args: ['class.igx-date-picker']
        }], dropdownCSS: [{
            type: HostBinding,
            args: ['class.igx-date-picker--dropdown']
        }], verticalCSS: [{
            type: HostBinding,
            args: ['class.igx-date-picker--vertical']
        }], onEscape: [{
            type: HostListener,
            args: ['keydown.esc', ['$event']]
        }, {
            type: HostListener,
            args: ['keydown.alt.arrowup', ['$event']]
        }], calendar: [{
            type: ViewChild,
            args: ['calendar', { static: true }]
        }], cancelButtonLabel: [{
            type: Input
        }], todayButtonLabel: [{
            type: Input
        }], datePickerActions: [{
            type: Input
        }] }); })();
