// igniteui-angular/esm2015/lib/date-picker/date-picker.pipes.js
import { Pipe, Inject } from '@angular/core';
import { IGX_DATE_PICKER_COMPONENT } from './date-picker.common';
import { DatePickerUtil } from './date-picker.utils';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class DatePickerDisplayValuePipe {
    constructor(_datePicker) {
        this._datePicker = _datePicker;
    }
    transform(value, args) {
        if (value !== '') {
            if (value === DatePickerUtil.maskToPromptChars(this._datePicker.inputMask)) {
                return '';
            }
            this._datePicker.rawDateString = value;
            return DatePickerUtil.trimEmptyPlaceholders(value);
        }
        return '';
    }
}
DatePickerDisplayValuePipe.ɵfac = function DatePickerDisplayValuePipe_Factory(t) { return new (t || DatePickerDisplayValuePipe)(ɵngcc0.ɵɵdirectiveInject(IGX_DATE_PICKER_COMPONENT)); };
DatePickerDisplayValuePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "displayValue", type: DatePickerDisplayValuePipe, pure: true });
DatePickerDisplayValuePipe.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [IGX_DATE_PICKER_COMPONENT,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatePickerDisplayValuePipe, [{
        type: Pipe,
        args: [{
                name: 'displayValue'
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [IGX_DATE_PICKER_COMPONENT]
            }] }]; }, null); })();
/**
 * @hidden
 */
export class DatePickerInputValuePipe {
    constructor(_datePicker) {
        this._datePicker = _datePicker;
    }
    transform(value, args) {
        /**
         * TODO(D.P.): This plugs into the mask, but constantly received display strings it can't handle at all
         * Those are almost immediately overridden by the pickers onFocus handling anyway; Refactor ASAP
         */
        if (this._datePicker.invalidDate !== '') {
            return this._datePicker.invalidDate;
        }
        else {
            if (this._datePicker.value === null || this._datePicker.value === undefined) {
                return DatePickerUtil.maskToPromptChars(this._datePicker.inputMask);
            }
            else {
                return this._datePicker._getEditorDate(this._datePicker.value);
                // return DatePickerUtil.addPromptCharsEditMode(this._datePicker.dateFormatParts, this._datePicker.value, value);
            }
        }
    }
}
DatePickerInputValuePipe.ɵfac = function DatePickerInputValuePipe_Factory(t) { return new (t || DatePickerInputValuePipe)(ɵngcc0.ɵɵdirectiveInject(IGX_DATE_PICKER_COMPONENT)); };
DatePickerInputValuePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "inputValue", type: DatePickerInputValuePipe, pure: true });
DatePickerInputValuePipe.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [IGX_DATE_PICKER_COMPONENT,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatePickerInputValuePipe, [{
        type: Pipe,
        args: [{
                name: 'inputValue'
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [IGX_DATE_PICKER_COMPONENT]
            }] }]; }, null); })();
