// igniteui-angular/esm2015/lib/date-picker/date-picker.directives.js
import { Directive, TemplateRef } from '@angular/core';
/**
 * IgxDatePickerTemplateDirective can be used to re-template the date-picker input-group.
 *
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxDatePickerTemplateDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxDatePickerTemplateDirective.ɵfac = function IgxDatePickerTemplateDirective_Factory(t) { return new (t || IgxDatePickerTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxDatePickerTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxDatePickerTemplateDirective, selectors: [["", "igxDatePickerTemplate", ""]] });
IgxDatePickerTemplateDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxDatePickerTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxDatePickerTemplate]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
/**
 * IgxDatePickerActionsDirective can be used to re-template the dropdown/dialog action buttons.
 */
export class IgxDatePickerActionsDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxDatePickerActionsDirective.ɵfac = function IgxDatePickerActionsDirective_Factory(t) { return new (t || IgxDatePickerActionsDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxDatePickerActionsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxDatePickerActionsDirective, selectors: [["", "igxDatePickerActions", ""]] });
IgxDatePickerActionsDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxDatePickerActionsDirective, [{
        type: Directive,
        args: [{
                selector: '[igxDatePickerActions]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
