// igniteui-angular/esm2015/lib/slider/slider.common.js
import { Directive } from '@angular/core';
/**
 * Template directive that allows you to set a custom template representing the lower label value of the {@link IgxSliderComponent}
 *
 * ```html
 * <igx-slider>
 *  <ng-template igxSliderThumbFrom let-value let-labels>{{value}}</ng-template>
 * </igx-slider>
 * ```
 *
 * @context {@link IgxSliderComponent.context}
 */
import * as ɵngcc0 from '@angular/core';
export class IgxThumbFromTemplateDirective {
}
IgxThumbFromTemplateDirective.ɵfac = function IgxThumbFromTemplateDirective_Factory(t) { return new (t || IgxThumbFromTemplateDirective)(); };
IgxThumbFromTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxThumbFromTemplateDirective, selectors: [["", "igxSliderThumbFrom", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxThumbFromTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxSliderThumbFrom]'
            }]
    }], null, null); })();
/**
 * Template directive that allows you to set a custom template representing the upper label value of the {@link IgxSliderComponent}
 *
 * ```html
 * <igx-slider>
 *  <ng-template igxSliderThumbTo let-value let-labels>{{value}}</ng-template>
 * </igx-slider>
 * ```
 *
 * @context {@link IgxSliderComponent.context}
 */
export class IgxThumbToTemplateDirective {
}
IgxThumbToTemplateDirective.ɵfac = function IgxThumbToTemplateDirective_Factory(t) { return new (t || IgxThumbToTemplateDirective)(); };
IgxThumbToTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxThumbToTemplateDirective, selectors: [["", "igxSliderThumbTo", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxThumbToTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxSliderThumbTo]'
            }]
    }], null, null); })();
/**
 * Template directive that allows you to set a custom template, represeting primary/secondary tick labels of the {@link IgxSliderComponent}
 *
 * @context {@link IgxTicksComponent.context}
 */
export class IgxTickLabelTemplateDirective {
}
IgxTickLabelTemplateDirective.ɵfac = function IgxTickLabelTemplateDirective_Factory(t) { return new (t || IgxTickLabelTemplateDirective)(); };
IgxTickLabelTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxTickLabelTemplateDirective, selectors: [["", "igxSliderTickLabel", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxTickLabelTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[igxSliderTickLabel]'
            }]
    }], null, null); })();
export var IgxSliderType;
(function (IgxSliderType) {
    /**
     * Slider with single thumb.
     */
    IgxSliderType[IgxSliderType["SLIDER"] = 0] = "SLIDER";
    /**
     *  Range slider with multiple thumbs, that can mark the range.
     */
    IgxSliderType[IgxSliderType["RANGE"] = 1] = "RANGE";
})(IgxSliderType || (IgxSliderType = {}));
export var SliderHandle;
(function (SliderHandle) {
    SliderHandle[SliderHandle["FROM"] = 0] = "FROM";
    SliderHandle[SliderHandle["TO"] = 1] = "TO";
})(SliderHandle || (SliderHandle = {}));
/**
 * Slider Tick labels Orientation
 */
export var TickLabelsOrientation;
(function (TickLabelsOrientation) {
    TickLabelsOrientation[TickLabelsOrientation["Horizontal"] = 0] = "Horizontal";
    TickLabelsOrientation[TickLabelsOrientation["TopToBottom"] = 1] = "TopToBottom";
    TickLabelsOrientation[TickLabelsOrientation["BottomToTop"] = 2] = "BottomToTop";
})(TickLabelsOrientation || (TickLabelsOrientation = {}));
/**
 * Slider Ticks orientation
 */
export var TicksOrientation;
(function (TicksOrientation) {
    TicksOrientation[TicksOrientation["Top"] = 0] = "Top";
    TicksOrientation[TicksOrientation["Bottom"] = 1] = "Bottom";
    TicksOrientation[TicksOrientation["Mirror"] = 2] = "Mirror";
})(TicksOrientation || (TicksOrientation = {}));
