// igniteui-angular/esm2015/lib/directives/label/label.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let NEXT_ID = 0;
export class IgxLabelDirective {
    constructor() {
        this.defaultClass = true;
        /**
         * @hidden
         */
        this.id = `igx-label-${NEXT_ID++}`;
    }
}
IgxLabelDirective.ɵfac = function IgxLabelDirective_Factory(t) { return new (t || IgxLabelDirective)(); };
IgxLabelDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxLabelDirective, selectors: [["", "igxLabel", ""]], hostVars: 3, hostBindings: function IgxLabelDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("id", ctx.id);
        ɵngcc0.ɵɵclassProp("igx-input-group__label", ctx.defaultClass);
    } }, inputs: { id: "id" } });
IgxLabelDirective.propDecorators = {
    defaultClass: [{ type: HostBinding, args: ['class.igx-input-group__label',] }],
    id: [{ type: HostBinding, args: ['attr.id',] }, { type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxLabelDirective, [{
        type: Directive,
        args: [{
                selector: '[igxLabel]'
            }]
    }], function () { return []; }, { defaultClass: [{
            type: HostBinding,
            args: ['class.igx-input-group__label']
        }], id: [{
            type: HostBinding,
            args: ['attr.id']
        }, {
            type: Input
        }] }); })();
