// igniteui-angular/esm2015/lib/directives/suffix/suffix.directive.js
import { NgModule, Directive } from '@angular/core';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
export class IgxSuffixDirective {
}
IgxSuffixDirective.ɵfac = function IgxSuffixDirective_Factory(t) { return new (t || IgxSuffixDirective)(); };
IgxSuffixDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxSuffixDirective, selectors: [["igx-suffix"], ["", "igxSuffix", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxSuffixDirective, [{
        type: Directive,
        args: [{
                selector: 'igx-suffix,[igxSuffix]'
            }]
    }], null, null); })();
/**
 * @hidden
 */
export class IgxSuffixModule {
}
IgxSuffixModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxSuffixModule });
IgxSuffixModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxSuffixModule_Factory(t) { return new (t || IgxSuffixModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxSuffixModule, { declarations: [IgxSuffixDirective], exports: [IgxSuffixDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxSuffixModule, [{
        type: NgModule,
        args: [{
                declarations: [IgxSuffixDirective],
                exports: [IgxSuffixDirective]
            }]
    }], null, null); })();
