// igniteui-angular/esm2015/lib/drop-down/public_api.js
import { NgModule } from '@angular/core';
import { IgxDropDownComponent } from './drop-down.component';
import { IgxDropDownItemComponent } from './drop-down-item.component';
import { IgxDropDownItemNavigationDirective } from './drop-down-navigation.directive';
import { CommonModule } from '@angular/common';
import { IgxToggleModule } from '../directives/toggle/toggle.directive';
import { IgxDropDownGroupComponent } from './drop-down-group.component';
import { IgxDropDownItemBaseDirective } from './drop-down-item.base';
import * as ɵngcc0 from '@angular/core';
export * from './drop-down.component';
export * from './drop-down-item.component';
export * from './drop-down-navigation.directive';
export * from './drop-down.base';
export * from './drop-down-item.base';
export * from './drop-down-group.component';
/**
 * @hidden
 */
export class IgxDropDownModule {
}
IgxDropDownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxDropDownModule });
IgxDropDownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxDropDownModule_Factory(t) { return new (t || IgxDropDownModule)(); }, imports: [[
            CommonModule,
            IgxToggleModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxDropDownModule, { declarations: function () { return [IgxDropDownComponent,
        IgxDropDownItemBaseDirective,
        IgxDropDownItemComponent,
        IgxDropDownGroupComponent,
        IgxDropDownItemNavigationDirective]; }, imports: function () { return [CommonModule,
        IgxToggleModule]; }, exports: function () { return [IgxDropDownComponent,
        IgxDropDownItemComponent,
        IgxDropDownGroupComponent,
        IgxDropDownItemNavigationDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxDropDownModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxDropDownComponent,
                    IgxDropDownItemBaseDirective,
                    IgxDropDownItemComponent,
                    IgxDropDownGroupComponent,
                    IgxDropDownItemNavigationDirective
                ],
                exports: [
                    IgxDropDownComponent,
                    IgxDropDownItemComponent,
                    IgxDropDownGroupComponent,
                    IgxDropDownItemNavigationDirective
                ],
                imports: [
                    CommonModule,
                    IgxToggleModule
                ]
            }]
    }], null, null); })();
