// igniteui-angular/esm2015/lib/grids/filtering/excel-style/excel-style-header.component.js
import { Component, Input } from '@angular/core';
import { IgxGridExcelStyleFilteringComponent } from './grid.excel-style-filtering.component';
/**
 * A component used for presenting Excel style header UI.
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './grid.excel-style-filtering.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../../directives/button/button.directive';
import * as ɵngcc4 from '../../../icon/icon.component';

function IgxExcelStyleHeaderComponent_header_0_button_4_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 5);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStyleHeaderComponent_header_0_button_4_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); const ctx_r4 = ɵngcc0.ɵɵnextContext(2); return ctx_r4.esf.onSelect(); });
    ɵngcc0.ɵɵelementStart(1, "igx-icon", 6);
    ɵngcc0.ɵɵtext(2, "done");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngClass", ctx_r1.esf.selectedClass());
} }
function IgxExcelStyleHeaderComponent_header_0_button_5_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 7);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStyleHeaderComponent_header_0_button_5_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r7); const ctx_r6 = ɵngcc0.ɵɵnextContext(2); return ctx_r6.esf.onPin(); });
    ɵngcc0.ɵɵelement(1, "igx-icon", 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("name", ctx_r2.esf.column.pinned ? "unpin-left" : "pin-left");
} }
function IgxExcelStyleHeaderComponent_header_0_button_6_Template(rf, ctx) { if (rf & 1) {
    const _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 7);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStyleHeaderComponent_header_0_button_6_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r9); const ctx_r8 = ɵngcc0.ɵɵnextContext(2); return ctx_r8.esf.onHideToggle(); });
    ɵngcc0.ɵɵelementStart(1, "igx-icon");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r3.esf.column.hidden ? "visibility" : "visibility_off");
} }
function IgxExcelStyleHeaderComponent_header_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "header", 1);
    ɵngcc0.ɵɵelementStart(1, "h4");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 2);
    ɵngcc0.ɵɵtemplate(4, IgxExcelStyleHeaderComponent_header_0_button_4_Template, 3, 1, "button", 3);
    ɵngcc0.ɵɵtemplate(5, IgxExcelStyleHeaderComponent_header_0_button_5_Template, 2, 1, "button", 4);
    ɵngcc0.ɵɵtemplate(6, IgxExcelStyleHeaderComponent_header_0_button_6_Template, 3, 1, "button", 4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.esf.column.header || ctx_r0.esf.column.field);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.showSelecting);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.showPinning);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.showHiding);
} }
export class IgxExcelStyleHeaderComponent {
    constructor(esf) {
        this.esf = esf;
    }
}
IgxExcelStyleHeaderComponent.ɵfac = function IgxExcelStyleHeaderComponent_Factory(t) { return new (t || IgxExcelStyleHeaderComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxGridExcelStyleFilteringComponent)); };
IgxExcelStyleHeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxExcelStyleHeaderComponent, selectors: [["igx-excel-style-header"]], inputs: { showPinning: "showPinning", showSelecting: "showSelecting", showHiding: "showHiding" }, decls: 1, vars: 1, consts: [["class", "igx-excel-filter__menu-header", 4, "ngIf"], [1, "igx-excel-filter__menu-header"], [1, "igx-excel-filter__menu-header-actions"], ["igxButton", "icon", 3, "ngClass", "click", 4, "ngIf"], ["igxButton", "icon", 3, "click", 4, "ngIf"], ["igxButton", "icon", 3, "ngClass", "click"], ["fontSet", "material"], ["igxButton", "icon", 3, "click"], ["fontSet", "imx-icons", 3, "name"]], template: function IgxExcelStyleHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxExcelStyleHeaderComponent_header_0_Template, 7, 4, "header", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.esf.column);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.IgxButtonDirective, ɵngcc2.NgClass, ɵngcc4.IgxIconComponent], encapsulation: 2 });
IgxExcelStyleHeaderComponent.ctorParameters = () => [
    { type: IgxGridExcelStyleFilteringComponent }
];
IgxExcelStyleHeaderComponent.propDecorators = {
    showPinning: [{ type: Input }],
    showSelecting: [{ type: Input }],
    showHiding: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExcelStyleHeaderComponent, [{
        type: Component,
        args: [{
                preserveWhitespaces: false,
                selector: 'igx-excel-style-header',
                template: "<header *ngIf=\"esf.column\"\r\n        class=\"igx-excel-filter__menu-header\">\r\n    <h4>{{ esf.column.header || esf.column.field }}</h4>\r\n    <div class=\"igx-excel-filter__menu-header-actions\">\r\n        <button *ngIf=\"showSelecting\"\r\n            igxButton=\"icon\"\r\n            (click)=\"esf.onSelect()\"\r\n            [ngClass]=\"esf.selectedClass()\"\r\n        >\r\n            <igx-icon fontSet=\"material\">done</igx-icon>\r\n        </button>\r\n        <button *ngIf=\"showPinning\"\r\n            igxButton=\"icon\"\r\n            (click)=\"esf.onPin()\"\r\n        >\r\n            <igx-icon fontSet=\"imx-icons\" [name]=\"esf.column.pinned ? 'unpin-left' : 'pin-left'\"></igx-icon>\r\n        </button>\r\n        <button *ngIf=\"showHiding\"\r\n            igxButton=\"icon\"\r\n            (click)=\"esf.onHideToggle()\"\r\n        >\r\n            <igx-icon>{{ esf.column.hidden ? 'visibility' : 'visibility_off' }}</igx-icon>\r\n        </button>\r\n    </div>\r\n</header>\r\n"
            }]
    }], function () { return [{ type: ɵngcc1.IgxGridExcelStyleFilteringComponent }]; }, { showPinning: [{
            type: Input
        }], showSelecting: [{
            type: Input
        }], showHiding: [{
            type: Input
        }] }); })();
