// igniteui-angular/esm2015/lib/grids/filtering/excel-style/excel-style-selecting.component.js
import { Component } from '@angular/core';
import { IgxGridExcelStyleFilteringComponent } from './grid.excel-style-filtering.component';
/**
 * A component used for presenting Excel style conditional filter UI.
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './grid.excel-style-filtering.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../../icon/icon.component';

function IgxExcelStyleSelectingComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r2 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStyleSelectingComponent_div_0_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r2); const ctx_r1 = ɵngcc0.ɵɵnextContext(); return ctx_r1.esf.onSelect(); });
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "igx-icon", 2);
    ɵngcc0.ɵɵtext(4, "done");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r0.esf.selectedClass());
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.esf.grid.resourceStrings.igx_grid_excel_select);
} }
export class IgxExcelStyleSelectingComponent {
    constructor(esf) {
        this.esf = esf;
    }
}
IgxExcelStyleSelectingComponent.ɵfac = function IgxExcelStyleSelectingComponent_Factory(t) { return new (t || IgxExcelStyleSelectingComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxGridExcelStyleFilteringComponent)); };
IgxExcelStyleSelectingComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxExcelStyleSelectingComponent, selectors: [["igx-excel-style-selecting"]], decls: 1, vars: 1, consts: [["tabindex", "0", 3, "ngClass", "click", 4, "ngIf"], ["tabindex", "0", 3, "ngClass", "click"], ["fontSet", "material"]], template: function IgxExcelStyleSelectingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxExcelStyleSelectingComponent_div_0_Template, 5, 2, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.esf.column);
    } }, directives: [ɵngcc2.NgIf, ɵngcc2.NgClass, ɵngcc3.IgxIconComponent], encapsulation: 2 });
IgxExcelStyleSelectingComponent.ctorParameters = () => [
    { type: IgxGridExcelStyleFilteringComponent }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExcelStyleSelectingComponent, [{
        type: Component,
        args: [{
                preserveWhitespaces: false,
                selector: 'igx-excel-style-selecting',
                template: "<div *ngIf=\"esf.column\"\n    [ngClass]=\"esf.selectedClass()\"\n    tabindex=\"0\"\n    (click)=\"esf.onSelect()\">\n    <span>{{esf.grid.resourceStrings.igx_grid_excel_select }}</span>\n    <igx-icon fontSet=\"material\">done</igx-icon>\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc1.IgxGridExcelStyleFilteringComponent }]; }, null); })();
