// igniteui-angular/esm2015/lib/grids/filtering/excel-style/excel-style-hiding.component.js
import { Component } from '@angular/core';
import { IgxGridExcelStyleFilteringComponent } from './grid.excel-style-filtering.component';
/**
 * A component used for presenting Excel style column hiding UI.
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './grid.excel-style-filtering.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../../icon/icon.component';

function IgxExcelStyleHidingComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r2 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStyleHidingComponent_div_0_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r2); const ctx_r1 = ɵngcc0.ɵɵnextContext(); return ctx_r1.esf.onHideToggle(); });
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "igx-icon");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.esf.column.hidden ? ctx_r0.esf.grid.resourceStrings.igx_grid_excel_show : ctx_r0.esf.grid.resourceStrings.igx_grid_excel_hide);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.esf.column.hidden ? "visibility" : "visibility_off");
} }
export class IgxExcelStyleHidingComponent {
    constructor(esf) {
        this.esf = esf;
    }
}
IgxExcelStyleHidingComponent.ɵfac = function IgxExcelStyleHidingComponent_Factory(t) { return new (t || IgxExcelStyleHidingComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxGridExcelStyleFilteringComponent)); };
IgxExcelStyleHidingComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxExcelStyleHidingComponent, selectors: [["igx-excel-style-hiding"]], decls: 1, vars: 1, consts: [["class", "igx-excel-filter__actions-hide", "tabindex", "0", 3, "click", 4, "ngIf"], ["tabindex", "0", 1, "igx-excel-filter__actions-hide", 3, "click"]], template: function IgxExcelStyleHidingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxExcelStyleHidingComponent_div_0_Template, 5, 2, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.esf.column);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.IgxIconComponent], encapsulation: 2 });
IgxExcelStyleHidingComponent.ctorParameters = () => [
    { type: IgxGridExcelStyleFilteringComponent }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExcelStyleHidingComponent, [{
        type: Component,
        args: [{
                preserveWhitespaces: false,
                selector: 'igx-excel-style-hiding',
                template: "<div *ngIf=\"esf.column\"\n    class=\"igx-excel-filter__actions-hide\"\n    tabindex=\"0\"\n    (click)=\"esf.onHideToggle()\">\n    <span>{{ esf.column.hidden ? esf.grid.resourceStrings.igx_grid_excel_show : esf.grid.resourceStrings.igx_grid_excel_hide }}</span>\n    <igx-icon>{{ esf.column.hidden ? 'visibility' : 'visibility_off' }}</igx-icon>\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc1.IgxGridExcelStyleFilteringComponent }]; }, null); })();
