// igniteui-angular/esm2015/lib/grids/filtering/excel-style/excel-style-pinning.component.js
import { Component } from '@angular/core';
import { IgxGridExcelStyleFilteringComponent } from './grid.excel-style-filtering.component';
/**
 * A component used for presenting Excel style column pinning UI.
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './grid.excel-style-filtering.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../../../icon/icon.component';

function IgxExcelStylePinningComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r2 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵlistener("click", function IgxExcelStylePinningComponent_div_0_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r2); const ctx_r1 = ɵngcc0.ɵɵnextContext(); return ctx_r1.esf.onPin(); });
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "igx-icon", 2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r0.esf.column.pinned ? "igx-excel-filter__actions-unpin" : "igx-excel-filter__actions-pin");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.esf.column.pinned ? ctx_r0.esf.grid.resourceStrings.igx_grid_excel_unpin : ctx_r0.esf.grid.resourceStrings.igx_grid_excel_pin);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("name", ctx_r0.esf.column.pinned ? "unpin-left" : "pin-left");
} }
export class IgxExcelStylePinningComponent {
    constructor(esf) {
        this.esf = esf;
    }
}
IgxExcelStylePinningComponent.ɵfac = function IgxExcelStylePinningComponent_Factory(t) { return new (t || IgxExcelStylePinningComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxGridExcelStyleFilteringComponent)); };
IgxExcelStylePinningComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxExcelStylePinningComponent, selectors: [["igx-excel-style-pinning"]], decls: 1, vars: 1, consts: [["tabindex", "0", 3, "ngClass", "click", 4, "ngIf"], ["tabindex", "0", 3, "ngClass", "click"], ["fontSet", "imx-icons", 3, "name"]], template: function IgxExcelStylePinningComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxExcelStylePinningComponent_div_0_Template, 4, 3, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.esf.column);
    } }, directives: [ɵngcc2.NgIf, ɵngcc2.NgClass, ɵngcc3.IgxIconComponent], encapsulation: 2 });
IgxExcelStylePinningComponent.ctorParameters = () => [
    { type: IgxGridExcelStyleFilteringComponent }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExcelStylePinningComponent, [{
        type: Component,
        args: [{
                preserveWhitespaces: false,
                selector: 'igx-excel-style-pinning',
                template: "<div *ngIf=\"esf.column\"\n    [ngClass]=\"esf.column.pinned ? 'igx-excel-filter__actions-unpin' : 'igx-excel-filter__actions-pin'\"\n    (click)=\"esf.onPin()\"\n    tabindex=\"0\">\n    <span>{{ esf.column.pinned ? esf.grid.resourceStrings.igx_grid_excel_unpin : esf.grid.resourceStrings.igx_grid_excel_pin }}</span>\n    <igx-icon fontSet=\"imx-icons\" name=\"{{ esf.column.pinned ? 'unpin-left' : 'pin-left' }}\"></igx-icon>\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc1.IgxGridExcelStyleFilteringComponent }]; }, null); })();
