// igniteui-angular/esm2015/lib/grids/filtering/excel-style/grid.excel-style-filtering.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IgxExcelStyleLoadingValuesTemplateDirective } from './excel-style-search.component';
import { IgxGridExcelStyleFilteringComponent, IgxExcelStyleColumnOperationsTemplateDirective, IgxExcelStyleFilterOperationsTemplateDirective } from './grid.excel-style-filtering.component';
import { IgxExcelStyleSortingComponent } from './excel-style-sorting.component';
import { IgxExcelStyleMovingComponent } from './excel-style-moving.component';
import { IgxExcelStyleSearchComponent } from './excel-style-search.component';
import { IgxExcelStyleCustomDialogComponent } from './excel-style-custom-dialog.component';
import { IgxExcelStyleDefaultExpressionComponent } from './excel-style-default-expression.component';
import { IgxExcelStyleDateExpressionComponent } from './excel-style-date-expression.component';
import { IgxSelectionAPIService } from '../../../core/selection';
import { FormsModule } from '@angular/forms';
import { IgxGridPipesModule } from '../../common/grid-pipes.module';
import { IgxButtonModule } from '../../../directives/button/button.directive';
import { IgxButtonGroupModule } from '../../../buttonGroup/buttonGroup.component';
import { IgxDatePickerModule } from '../../../date-picker/date-picker.component';
import { IgxIconModule } from '../../../icon/public_api';
import { IgxRippleModule } from '../../../directives/ripple/ripple.directive';
import { IgxInputGroupModule } from '../../../input-group/input-group.component';
import { IgxDropDownModule } from '../../../drop-down/public_api';
import { IgxForOfModule } from '../../../directives/for-of/for_of.directive';
import { IgxCheckboxModule } from '../../../checkbox/checkbox.component';
import { IgxFilterModule } from '../../../directives/filter/filter.directive';
import { IgxToggleModule } from '../../../directives/toggle/toggle.directive';
import { IgxListModule } from '../../../list/list.component';
import { IgxProgressBarModule } from '../../../progressbar/progressbar.component';
import { IgxSelectModule } from './../../../select/select.module';
import { IgxExcelStylePinningComponent } from './excel-style-pinning.component';
import { IgxExcelStyleHeaderComponent } from './excel-style-header.component';
import { IgxExcelStyleHidingComponent } from './excel-style-hiding.component';
import { IgxExcelStyleSelectingComponent } from './excel-style-selecting.component';
import { IgxExcelStyleClearFiltersComponent } from './excel-style-clear-filters.component';
import { IgxExcelStyleConditionalFilterComponent } from './excel-style-conditional-filter.component';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from '../../../directives/button/button.directive';
import * as ɵngcc4 from '../../../buttonGroup/buttonGroup.component';
import * as ɵngcc5 from '../../../date-picker/date-picker.component';
import * as ɵngcc6 from '../../../date-picker/date-picker.directives';
import * as ɵngcc7 from '../../../icon/icon.component';
import * as ɵngcc8 from '../../../directives/ripple/ripple.directive';
import * as ɵngcc9 from '../../../input-group/input-group.component';
import * as ɵngcc10 from '../../../directives/hint/hint.directive';
import * as ɵngcc11 from '../../../directives/input/input.directive';
import * as ɵngcc12 from '../../../directives/label/label.directive';
import * as ɵngcc13 from '../../../directives/prefix/prefix.directive';
import * as ɵngcc14 from '../../../directives/suffix/suffix.directive';
import * as ɵngcc15 from '../../../drop-down/drop-down.component';
import * as ɵngcc16 from '../../../drop-down/drop-down-item.component';
import * as ɵngcc17 from '../../../drop-down/drop-down-group.component';
import * as ɵngcc18 from '../../../drop-down/drop-down-navigation.directive';
import * as ɵngcc19 from '../../../directives/for-of/for_of.directive';
import * as ɵngcc20 from '../../../checkbox/checkbox.component';
import * as ɵngcc21 from '../../../directives/filter/filter.directive';
import * as ɵngcc22 from '../../../directives/toggle/toggle.directive';
import * as ɵngcc23 from '../../../list/list.component';
import * as ɵngcc24 from '../../../list/list-item.component';
import * as ɵngcc25 from '../../../list/list.common';
import * as ɵngcc26 from '../../../progressbar/progressbar.component';
import * as ɵngcc27 from '../../../progressbar/progressbar.common';
import * as ɵngcc28 from '../../../select/select.component';
import * as ɵngcc29 from '../../../select/select-group.component';
import * as ɵngcc30 from '../../../select/select-item.component';
import * as ɵngcc31 from '../../../select/select-navigation.directive';
import * as ɵngcc32 from '../../common/pipes';
import * as ɵngcc33 from '../../../date-picker/date-picker.pipes';
export class IgxGridExcelStyleFilteringModule {
}
IgxGridExcelStyleFilteringModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridExcelStyleFilteringModule });
IgxGridExcelStyleFilteringModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridExcelStyleFilteringModule_Factory(t) { return new (t || IgxGridExcelStyleFilteringModule)(); }, providers: [
        IgxSelectionAPIService
    ], imports: [[
            CommonModule,
            FormsModule,
            IgxGridPipesModule,
            IgxButtonModule,
            IgxButtonGroupModule,
            IgxDatePickerModule,
            IgxIconModule,
            IgxRippleModule,
            IgxInputGroupModule,
            IgxDropDownModule,
            IgxForOfModule,
            IgxCheckboxModule,
            IgxFilterModule,
            IgxToggleModule,
            IgxListModule,
            IgxProgressBarModule,
            IgxSelectModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridExcelStyleFilteringModule, { declarations: function () { return [IgxGridExcelStyleFilteringComponent,
        IgxExcelStyleHeaderComponent,
        IgxExcelStyleSortingComponent,
        IgxExcelStylePinningComponent,
        IgxExcelStyleHidingComponent,
        IgxExcelStyleSelectingComponent,
        IgxExcelStyleClearFiltersComponent,
        IgxExcelStyleConditionalFilterComponent,
        IgxExcelStyleMovingComponent,
        IgxExcelStyleSearchComponent,
        IgxExcelStyleCustomDialogComponent,
        IgxExcelStyleDefaultExpressionComponent,
        IgxExcelStyleDateExpressionComponent,
        IgxExcelStyleColumnOperationsTemplateDirective,
        IgxExcelStyleFilterOperationsTemplateDirective,
        IgxExcelStyleLoadingValuesTemplateDirective]; }, imports: function () { return [CommonModule,
        FormsModule,
        IgxGridPipesModule,
        IgxButtonModule,
        IgxButtonGroupModule,
        IgxDatePickerModule,
        IgxIconModule,
        IgxRippleModule,
        IgxInputGroupModule,
        IgxDropDownModule,
        IgxForOfModule,
        IgxCheckboxModule,
        IgxFilterModule,
        IgxToggleModule,
        IgxListModule,
        IgxProgressBarModule,
        IgxSelectModule]; }, exports: function () { return [IgxGridExcelStyleFilteringComponent,
        IgxExcelStyleColumnOperationsTemplateDirective,
        IgxExcelStyleFilterOperationsTemplateDirective,
        IgxExcelStyleLoadingValuesTemplateDirective,
        IgxExcelStyleDateExpressionComponent,
        IgxExcelStyleHeaderComponent,
        IgxExcelStyleSortingComponent,
        IgxExcelStylePinningComponent,
        IgxExcelStyleHidingComponent,
        IgxExcelStyleSelectingComponent,
        IgxExcelStyleClearFiltersComponent,
        IgxExcelStyleConditionalFilterComponent,
        IgxExcelStyleMovingComponent,
        IgxExcelStyleSearchComponent,
        IgxExcelStyleHeaderComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridExcelStyleFilteringModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxGridExcelStyleFilteringComponent,
                    IgxExcelStyleHeaderComponent,
                    IgxExcelStyleSortingComponent,
                    IgxExcelStylePinningComponent,
                    IgxExcelStyleHidingComponent,
                    IgxExcelStyleSelectingComponent,
                    IgxExcelStyleClearFiltersComponent,
                    IgxExcelStyleConditionalFilterComponent,
                    IgxExcelStyleMovingComponent,
                    IgxExcelStyleSearchComponent,
                    IgxExcelStyleCustomDialogComponent,
                    IgxExcelStyleDefaultExpressionComponent,
                    IgxExcelStyleDateExpressionComponent,
                    IgxExcelStyleColumnOperationsTemplateDirective,
                    IgxExcelStyleFilterOperationsTemplateDirective,
                    IgxExcelStyleLoadingValuesTemplateDirective
                ],
                exports: [
                    IgxGridExcelStyleFilteringComponent,
                    IgxExcelStyleColumnOperationsTemplateDirective,
                    IgxExcelStyleFilterOperationsTemplateDirective,
                    IgxExcelStyleLoadingValuesTemplateDirective,
                    IgxExcelStyleDateExpressionComponent,
                    IgxExcelStyleHeaderComponent,
                    IgxExcelStyleSortingComponent,
                    IgxExcelStylePinningComponent,
                    IgxExcelStyleHidingComponent,
                    IgxExcelStyleSelectingComponent,
                    IgxExcelStyleClearFiltersComponent,
                    IgxExcelStyleConditionalFilterComponent,
                    IgxExcelStyleMovingComponent,
                    IgxExcelStyleSearchComponent,
                    IgxExcelStyleHeaderComponent
                ],
                imports: [
                    CommonModule,
                    FormsModule,
                    IgxGridPipesModule,
                    IgxButtonModule,
                    IgxButtonGroupModule,
                    IgxDatePickerModule,
                    IgxIconModule,
                    IgxRippleModule,
                    IgxInputGroupModule,
                    IgxDropDownModule,
                    IgxForOfModule,
                    IgxCheckboxModule,
                    IgxFilterModule,
                    IgxToggleModule,
                    IgxListModule,
                    IgxProgressBarModule,
                    IgxSelectModule,
                ],
                entryComponents: [
                    IgxGridExcelStyleFilteringComponent
                ],
                providers: [
                    IgxSelectionAPIService
                ]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(IgxGridExcelStyleFilteringComponent, [ɵngcc1.NgClass, ɵngcc1.NgComponentOutlet, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgTemplateOutlet, ɵngcc1.NgStyle, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc1.NgPlural, ɵngcc1.NgPluralCase, ɵngcc2.ɵangular_packages_forms_forms_y, ɵngcc2.NgSelectOption, ɵngcc2.ɵangular_packages_forms_forms_x, ɵngcc2.DefaultValueAccessor, ɵngcc2.NumberValueAccessor, ɵngcc2.RangeValueAccessor, ɵngcc2.CheckboxControlValueAccessor, ɵngcc2.SelectControlValueAccessor, ɵngcc2.SelectMultipleControlValueAccessor, ɵngcc2.RadioControlValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgControlStatusGroup, ɵngcc2.RequiredValidator, ɵngcc2.MinLengthValidator, ɵngcc2.MaxLengthValidator, ɵngcc2.PatternValidator, ɵngcc2.CheckboxRequiredValidator, ɵngcc2.EmailValidator, ɵngcc2.NgModel, ɵngcc2.NgModelGroup, ɵngcc2.NgForm, ɵngcc3.IgxButtonDirective, ɵngcc4.IgxButtonGroupComponent, ɵngcc5.IgxDatePickerComponent, ɵngcc6.IgxDatePickerTemplateDirective, ɵngcc6.IgxDatePickerActionsDirective, ɵngcc7.IgxIconComponent, ɵngcc8.IgxRippleDirective, ɵngcc9.IgxInputGroupComponent, ɵngcc10.IgxHintDirective, ɵngcc11.IgxInputDirective, ɵngcc12.IgxLabelDirective, ɵngcc13.IgxPrefixDirective, ɵngcc14.IgxSuffixDirective, ɵngcc15.IgxDropDownComponent, ɵngcc16.IgxDropDownItemComponent, ɵngcc17.IgxDropDownGroupComponent, ɵngcc18.IgxDropDownItemNavigationDirective, ɵngcc19.IgxForOfDirective, ɵngcc19.IgxGridForOfDirective, ɵngcc20.IgxCheckboxComponent, ɵngcc20.IgxCheckboxRequiredDirective, ɵngcc21.IgxFilterDirective, ɵngcc22.IgxToggleDirective, ɵngcc22.IgxToggleActionDirective, ɵngcc22.IgxOverlayOutletDirective, ɵngcc23.IgxListComponent, ɵngcc24.IgxListItemComponent, ɵngcc23.IgxListThumbnailDirective, ɵngcc23.IgxListActionDirective, ɵngcc23.IgxListLineDirective, ɵngcc23.IgxListLineTitleDirective, ɵngcc23.IgxListLineSubTitleDirective, ɵngcc25.IgxDataLoadingTemplateDirective, ɵngcc25.IgxEmptyListTemplateDirective, ɵngcc25.IgxListItemLeftPanningTemplateDirective, ɵngcc25.IgxListItemRightPanningTemplateDirective, ɵngcc26.IgxLinearProgressBarComponent, ɵngcc26.IgxCircularProgressBarComponent, ɵngcc27.IgxProcessBarTextTemplateDirective, ɵngcc27.IgxProgressBarGradientDirective, ɵngcc28.IgxSelectComponent, ɵngcc28.IgxSelectFooterDirective, ɵngcc29.IgxSelectGroupComponent, ɵngcc28.IgxSelectHeaderDirective, ɵngcc30.IgxSelectItemComponent, ɵngcc31.IgxSelectItemNavigationDirective, ɵngcc28.IgxSelectToggleIconDirective, IgxGridExcelStyleFilteringComponent,
    IgxExcelStyleHeaderComponent,
    IgxExcelStyleSortingComponent,
    IgxExcelStylePinningComponent,
    IgxExcelStyleHidingComponent,
    IgxExcelStyleSelectingComponent,
    IgxExcelStyleClearFiltersComponent,
    IgxExcelStyleConditionalFilterComponent,
    IgxExcelStyleMovingComponent,
    IgxExcelStyleSearchComponent,
    IgxExcelStyleCustomDialogComponent,
    IgxExcelStyleDefaultExpressionComponent,
    IgxExcelStyleDateExpressionComponent,
    IgxExcelStyleColumnOperationsTemplateDirective,
    IgxExcelStyleFilterOperationsTemplateDirective,
    IgxExcelStyleLoadingValuesTemplateDirective], [ɵngcc1.AsyncPipe, ɵngcc1.UpperCasePipe, ɵngcc1.LowerCasePipe, ɵngcc1.JsonPipe, ɵngcc1.SlicePipe, ɵngcc1.DecimalPipe, ɵngcc1.PercentPipe, ɵngcc1.TitleCasePipe, ɵngcc1.CurrencyPipe, ɵngcc1.DatePipe, ɵngcc1.I18nPluralPipe, ɵngcc1.I18nSelectPipe, ɵngcc1.KeyValuePipe, ɵngcc32.IgxGridFilterConditionPipe, ɵngcc32.IgxGridTransactionPipe, ɵngcc32.IgxGridNotGroupedPipe, ɵngcc32.IgxGridTopLevelColumns, ɵngcc32.IgxGridCellStylesPipe, ɵngcc32.IgxGridCellStyleClassesPipe, ɵngcc32.IgxGridPaginatorOptionsPipe, ɵngcc32.IgxHasVisibleColumnsPipe, ɵngcc32.IgxGridRowPinningPipe, ɵngcc32.IgxColumnActionEnabledPipe, ɵngcc32.IgxFilterActionColumnsPipe, ɵngcc32.IgxSortActionColumnsPipe, ɵngcc32.IgxGridDataMapperPipe, ɵngcc32.IgxStringReplacePipe, ɵngcc32.IgxGridTransactionStatePipe, ɵngcc32.IgxGridAddRowPipe, ɵngcc32.IgxColumnFormatterPipe, ɵngcc33.DatePickerDisplayValuePipe, ɵngcc33.DatePickerInputValuePipe, ɵngcc21.IgxFilterPipe]);
