// igniteui-angular/esm2015/lib/splitter/splitter.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IgxSplitBarComponent } from './splitter-bar/splitter-bar.component';
import { IgxSplitterPaneComponent } from './splitter-pane/splitter-pane.component';
import { IgxSplitterComponent } from './splitter.component';
import { IgxIconModule } from '../icon/public_api';
import { IgxDragDropModule } from '../directives/drag-drop/drag-drop.directive';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../icon/icon.component';
import * as ɵngcc3 from '../directives/drag-drop/drag-drop.directive';
export class IgxSplitterModule {
}
IgxSplitterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxSplitterModule });
IgxSplitterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxSplitterModule_Factory(t) { return new (t || IgxSplitterModule)(); }, imports: [[
            CommonModule, IgxIconModule, IgxDragDropModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxSplitterModule, { declarations: function () { return [IgxSplitterComponent,
        IgxSplitterPaneComponent,
        IgxSplitBarComponent]; }, imports: function () { return [CommonModule, IgxIconModule, IgxDragDropModule]; }, exports: function () { return [IgxSplitterComponent,
        IgxSplitterPaneComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxSplitterModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule, IgxIconModule, IgxDragDropModule
                ],
                declarations: [
                    IgxSplitterComponent,
                    IgxSplitterPaneComponent,
                    IgxSplitBarComponent
                ],
                exports: [
                    IgxSplitterComponent,
                    IgxSplitterPaneComponent
                ]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(IgxSplitterComponent, [ɵngcc1.NgClass, ɵngcc1.NgComponentOutlet, ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgTemplateOutlet, ɵngcc1.NgStyle, ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault, ɵngcc1.NgPlural, ɵngcc1.NgPluralCase, ɵngcc2.IgxIconComponent, ɵngcc3.IgxDragDirective, ɵngcc3.IgxDropDirective, ɵngcc3.IgxDragHandleDirective, ɵngcc3.IgxDragIgnoreDirective, IgxSplitterComponent,
    IgxSplitterPaneComponent,
    IgxSplitBarComponent], [ɵngcc1.AsyncPipe, ɵngcc1.UpperCasePipe, ɵngcc1.LowerCasePipe, ɵngcc1.JsonPipe, ɵngcc1.SlicePipe, ɵngcc1.DecimalPipe, ɵngcc1.PercentPipe, ɵngcc1.TitleCasePipe, ɵngcc1.CurrencyPipe, ɵngcc1.DatePipe, ɵngcc1.I18nPluralPipe, ɵngcc1.I18nSelectPipe, ɵngcc1.KeyValuePipe]);
