// igniteui-angular/esm2015/lib/grids/resizing/resize.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IgxColumnResizingService } from './resizing.service';
import { IgxGridColumnResizerComponent } from './resizer.component';
import { IgxResizeHandleDirective } from './resize-handle.directive';
import { IgxColumnResizerDirective } from './resizer.directive';
import * as ɵngcc0 from '@angular/core';
export class IgxGridResizingModule {
}
IgxGridResizingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridResizingModule });
IgxGridResizingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridResizingModule_Factory(t) { return new (t || IgxGridResizingModule)(); }, providers: [
        IgxColumnResizingService
    ], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridResizingModule, { declarations: function () { return [IgxGridColumnResizerComponent,
        IgxResizeHandleDirective,
        IgxColumnResizerDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [IgxGridColumnResizerComponent,
        IgxResizeHandleDirective,
        IgxColumnResizerDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridResizingModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxGridColumnResizerComponent,
                    IgxResizeHandleDirective,
                    IgxColumnResizerDirective
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    IgxGridColumnResizerComponent,
                    IgxResizeHandleDirective,
                    IgxColumnResizerDirective
                ],
                providers: [
                    IgxColumnResizingService
                ]
            }]
    }], null, null); })();
