// igniteui-angular/esm2015/lib/grids/resizing/resizer.component.js
import { ChangeDetectionStrategy, Component, ViewChild } from '@angular/core';
import { IgxColumnResizingService } from './resizing.service';
import { IgxColumnResizerDirective } from './resizer.directive';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './resizing.service';
import * as ɵngcc2 from './resizer.directive';
export class IgxGridColumnResizerComponent {
    constructor(colResizingService) {
        this.colResizingService = colResizingService;
    }
}
IgxGridColumnResizerComponent.ɵfac = function IgxGridColumnResizerComponent_Factory(t) { return new (t || IgxGridColumnResizerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxColumnResizingService)); };
IgxGridColumnResizerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxGridColumnResizerComponent, selectors: [["igx-grid-column-resizer"]], viewQuery: function IgxGridColumnResizerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(IgxColumnResizerDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.resizer = _t.first);
    } }, decls: 1, vars: 6, consts: [["igxResizer", "", 1, "igx-grid__th-resize-line", 3, "restrictHResizeMax", "restrictHResizeMin", "resizeEnd"]], template: function IgxGridColumnResizerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵlistener("resizeEnd", function IgxGridColumnResizerComponent_Template_div_resizeEnd_0_listener($event) { return ctx.colResizingService.resizeColumn($event); });
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("left", -99999, "px")("height", ctx.colResizingService.resizerHeight, "px");
        ɵngcc0.ɵɵproperty("restrictHResizeMax", ctx.colResizingService.restrictResizeMax)("restrictHResizeMin", ctx.colResizingService.restrictResizeMin);
    } }, directives: [ɵngcc2.IgxColumnResizerDirective], encapsulation: 2, changeDetection: 0 });
IgxGridColumnResizerComponent.ctorParameters = () => [
    { type: IgxColumnResizingService }
];
IgxGridColumnResizerComponent.propDecorators = {
    resizer: [{ type: ViewChild, args: [IgxColumnResizerDirective, { static: true },] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridColumnResizerComponent, [{
        type: Component,
        args: [{
                changeDetection: ChangeDetectionStrategy.OnPush,
                preserveWhitespaces: false,
                selector: 'igx-grid-column-resizer',
                template: "<div class=\"igx-grid__th-resize-line\"\n    [style.left.px]=\"-99999\" igxResizer\n    [style.height.px]=\"colResizingService.resizerHeight\"\n    [restrictHResizeMax]=\"colResizingService.restrictResizeMax\"\n    [restrictHResizeMin]=\"colResizingService.restrictResizeMin\"\n    (resizeEnd)=\"colResizingService.resizeColumn($event)\">\n</div>\n"
            }]
    }], function () { return [{ type: ɵngcc1.IgxColumnResizingService }]; }, { resizer: [{
            type: ViewChild,
            args: [IgxColumnResizerDirective, { static: true }]
        }] }); })();
