// igniteui-angular/esm2015/lib/calendar/months-view.pipe.js
import { Pipe } from '@angular/core';
import { Calendar } from './calendar';
import * as ɵngcc0 from '@angular/core';
export class IgxMonthViewSlotsCalendar {
    transform(monthViews) {
        return new Array(monthViews);
    }
}
IgxMonthViewSlotsCalendar.ɵfac = function IgxMonthViewSlotsCalendar_Factory(t) { return new (t || IgxMonthViewSlotsCalendar)(); };
IgxMonthViewSlotsCalendar.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "IgxMonthViewSlots", type: IgxMonthViewSlotsCalendar, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxMonthViewSlotsCalendar, [{
        type: Pipe,
        args: [{
                name: 'IgxMonthViewSlots'
            }]
    }], null, null); })();
export class IgxGetViewDateCalendar {
    constructor() {
        this.calendar = new Calendar();
    }
    transform(index, viewDate, wholeDate = true) {
        const date = this.calendar.timedelta(viewDate, 'month', index);
        return wholeDate ? date : date.getMonth();
    }
}
IgxGetViewDateCalendar.ɵfac = function IgxGetViewDateCalendar_Factory(t) { return new (t || IgxGetViewDateCalendar)(); };
IgxGetViewDateCalendar.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "IgxGetViewDate", type: IgxGetViewDateCalendar, pure: true });
IgxGetViewDateCalendar.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGetViewDateCalendar, [{
        type: Pipe,
        args: [{
                name: 'IgxGetViewDate'
            }]
    }], function () { return []; }, null); })();
