// igniteui-angular/esm2015/lib/action-strip/grid-actions/grid-editing-actions.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { IgxGridActionsBaseDirective } from './grid-actions-base.directive';
import { showMessage } from '../../core/deprecateDecorators';
import { addRow, addChild } from '@igniteui/material-icons-extended';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from './grid-action-button.component';

function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_1_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 5);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_1_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r6); const ctx_r5 = ɵngcc0.ɵɵnextContext(2); return ctx_r5.startEdit($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r1.asMenuItems)("labelText", ctx_r1.grid.resourceStrings.igx_grid_actions_edit_label);
} }
function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 6);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_2_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r8); const ctx_r7 = ɵngcc0.ɵɵnextContext(2); return ctx_r7.addRowHandler($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r2.asMenuItems)("labelText", ctx_r2.grid.resourceStrings.igx_grid_actions_add_label);
} }
function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_3_Template(rf, ctx) { if (rf & 1) {
    const _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 7);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_3_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); const ctx_r9 = ɵngcc0.ɵɵnextContext(2); return ctx_r9.addRowHandler($event, true); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r3.asMenuItems)("labelText", ctx_r3.grid.resourceStrings.igx_grid_actions_add_child_label);
} }
function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_4_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 8);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_4_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.deleteRow($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r4.asMenuItems)("labelText", ctx_r4.grid.resourceStrings.igx_grid_actions_delete_label);
} }
function IgxGridEditingActionsComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_1_Template, 1, 2, "igx-grid-action-button", 1);
    ɵngcc0.ɵɵtemplate(2, IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_2_Template, 1, 2, "igx-grid-action-button", 2);
    ɵngcc0.ɵɵtemplate(3, IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_3_Template, 1, 2, "igx-grid-action-button", 3);
    ɵngcc0.ɵɵtemplate(4, IgxGridEditingActionsComponent_ng_container_0_igx_grid_action_button_4_Template, 1, 2, "igx-grid-action-button", 4);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.disabled);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.addRow && ctx_r0.isRootRow);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.addChild && ctx_r0.hasChildren);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.disabled);
} }
export class IgxGridEditingActionsComponent extends IgxGridActionsBaseDirective {
    constructor() {
        super(...arguments);
        /**
         * Host `class.igx-action-strip` binding.
         * @hidden
         * @internal
         */
        this.cssClass = 'igx-action-strip__editing-actions';
        this.isMessageShown = false;
        this._addRow = false;
        this.iconsRendered = false;
        /**
         * An input to enable/disable action strip child row adding button
         */
        this.addChild = false;
    }
    /**
     * An input to enable/disable action strip row adding button
     */
    set addRow(value) {
        this._addRow = value;
    }
    get addRow() {
        if (!this.iconsRendered) {
            this.registerIcons();
            this.iconsRendered = true;
        }
        return this._addRow;
    }
    /**
     * Enter row or cell edit mode depending the grid rowEditable option
     * @example
     * ```typescript
     * this.gridEditingActions.startEdit();
     * ```
     */
    startEdit(event) {
        if (event) {
            event.stopPropagation();
        }
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const row = this.strip.context;
        const firstEditable = row.cells.filter(cell => cell.editable)[0];
        const grid = row.grid;
        if (!grid.hasEditableColumns) {
            this.isMessageShown = showMessage('The grid should be editable in order to use IgxGridEditingActionsComponent', this.isMessageShown);
            return;
        }
        // be sure row is in view
        if (grid.rowList.filter(r => r === row).length !== 0) {
            grid.crudService.enterEditMode(firstEditable);
        }
        this.strip.hide();
    }
    /**
     * Delete a row according to the context
     * @example
     * ```typescript
     * this.gridEditingActions.deleteRow();
     * ```
     */
    deleteRow(event) {
        if (event) {
            event.stopPropagation();
        }
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const context = this.strip.context;
        const grid = context.grid;
        grid.deleteRow(context.rowID);
        this.strip.hide();
    }
    addRowHandler(event, asChild) {
        if (event) {
            event.stopPropagation();
        }
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const context = this.strip.context;
        const grid = context.grid;
        if (!grid.rowEditable) {
            console.warn('The grid must use row edit mode to perform row adding! Please set rowEditable to true.');
            return;
        }
        grid.beginAddRowByIndex(context.rowID, context.index, asChild);
        this.strip.hide();
    }
    /**
     * Getter if the row is disabled
     * @hidden
     * @internal
     */
    get disabled() {
        if (!this.isRow(this.strip.context)) {
            return;
        }
        return this.strip.context.disabled;
    }
    /**
     * Getter if the row is root.
     * @hidden
     * @internal
     */
    get isRootRow() {
        if (!this.isRow(this.strip.context)) {
            return false;
        }
        return this.strip.context.isRoot;
    }
    get hasChildren() {
        if (!this.isRow(this.strip.context)) {
            return false;
        }
        return this.strip.context.hasChildren;
    }
    /**
     * @hidden
     * @internal
     */
    registerIcons() {
        this.iconService.addSvgIconFromText(addRow.name, addRow.value, 'imx-icons');
        this.iconService.addSvgIconFromText(addChild.name, addChild.value, 'imx-icons');
    }
}
IgxGridEditingActionsComponent.ɵfac = function IgxGridEditingActionsComponent_Factory(t) { return ɵIgxGridEditingActionsComponent_BaseFactory(t || IgxGridEditingActionsComponent); };
IgxGridEditingActionsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxGridEditingActionsComponent, selectors: [["igx-grid-editing-actions"]], hostVars: 2, hostBindings: function IgxGridEditingActionsComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-action-strip__editing-actions", ctx.cssClass);
    } }, inputs: { addChild: "addChild", addRow: "addRow" }, features: [ɵngcc0.ɵɵProvidersFeature([{ provide: IgxGridActionsBaseDirective, useExisting: IgxGridEditingActionsComponent }]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [[4, "ngIf"], ["iconName", "edit", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconName", "add-row", "iconSet", "imx-icons", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconName", "add-child", "iconSet", "imx-icons", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["class", "igx-action-strip__delete", "classNames", "igx-action-strip__menu-item--danger", "iconName", "delete", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconName", "edit", 3, "asMenuItem", "labelText", "onActionClick"], ["iconName", "add-row", "iconSet", "imx-icons", 3, "asMenuItem", "labelText", "onActionClick"], ["iconName", "add-child", "iconSet", "imx-icons", 3, "asMenuItem", "labelText", "onActionClick"], ["classNames", "igx-action-strip__menu-item--danger", "iconName", "delete", 1, "igx-action-strip__delete", 3, "asMenuItem", "labelText", "onActionClick"]], template: function IgxGridEditingActionsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxGridEditingActionsComponent_ng_container_0_Template, 5, 4, "ng-container", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRowContext);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.IgxGridActionButtonComponent], encapsulation: 2 });
IgxGridEditingActionsComponent.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-action-strip__editing-actions',] }],
    addRow: [{ type: Input }],
    addChild: [{ type: Input }]
};
const ɵIgxGridEditingActionsComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(IgxGridEditingActionsComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridEditingActionsComponent, [{
        type: Component,
        args: [{
                selector: 'igx-grid-editing-actions',
                template: "\n<ng-container *ngIf=\"isRowContext\">\n    <igx-grid-action-button *ngIf=\"!disabled\" [asMenuItem]=\"asMenuItems\" iconName=\"edit\" [labelText]=\"grid.resourceStrings.igx_grid_actions_edit_label\" (onActionClick)=\"startEdit($event)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"addRow && isRootRow\" [asMenuItem]=\"asMenuItems\" iconName=\"add-row\" iconSet=\"imx-icons\" [labelText]=\"grid.resourceStrings.igx_grid_actions_add_label\" (onActionClick)=\"addRowHandler($event)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"addChild && hasChildren\" [asMenuItem]=\"asMenuItems\" iconName=\"add-child\" iconSet=\"imx-icons\" [labelText]=\"grid.resourceStrings.igx_grid_actions_add_child_label\" (onActionClick)=\"addRowHandler($event, true)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"!disabled\" class=\"igx-action-strip__delete\" classNames='igx-action-strip__menu-item--danger' [asMenuItem]=\"asMenuItems\" iconName=\"delete\" [labelText]=\"grid.resourceStrings.igx_grid_actions_delete_label\" (onActionClick)=\"deleteRow($event)\"></igx-grid-action-button>\n</ng-container>\n\n",
                providers: [{ provide: IgxGridActionsBaseDirective, useExisting: IgxGridEditingActionsComponent }]
            }]
    }], null, { cssClass: [{
            type: HostBinding,
            args: ['class.igx-action-strip__editing-actions']
        }], addChild: [{
            type: Input
        }], addRow: [{
            type: Input
        }] }); })();
