// igniteui-angular/esm2015/lib/action-strip/grid-actions/grid-actions-base.directive.js
import { IgxGridActionButtonComponent } from './grid-action-button.component';
import { Directive, Input, ViewChildren, IterableDiffers } from '@angular/core';
import { IgxRowDirective } from '../../grids/row.directive';
import { IgxIconService } from '../../icon/icon.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../icon/icon.service';
export class IgxGridActionsBaseDirective {
    constructor(iconService, differs) {
        this.iconService = iconService;
        this.differs = differs;
        /**
         * Gets/Sets if the action buttons will be rendered as menu items. When in menu, items will be rendered with text label.
         * @example
         * ```html
         *  <igx-grid-pinning-actions [asMenuItems]='true'></igx-grid-pinning-actions>
         *  <igx-grid-editing-actions [asMenuItems]='true'></igx-grid-editing-actions>
         * ```
         */
        this.asMenuItems = false;
    }
    /**
     * @hidden
     * @internal
     */
    get grid() {
        return this.strip.context.grid;
    }
    /**
     * @hidden
     * @internal
     */
    ngAfterViewInit() {
        if (this.asMenuItems) {
            this.buttons.changes.subscribe((change) => {
                this.strip.cdr.detectChanges();
            });
        }
    }
    /**
     * Getter to be used in template
     * @hidden
     * @internal
     */
    get isRowContext() {
        return this.isRow(this.strip.context) && !this.strip.context.inEditMode;
    }
    /**
     * Check if the param is a row from a grid
     * @hidden
     * @internal
     * @param context
     */
    isRow(context) {
        return context && context instanceof IgxRowDirective;
    }
}
IgxGridActionsBaseDirective.ɵfac = function IgxGridActionsBaseDirective_Factory(t) { return new (t || IgxGridActionsBaseDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IgxIconService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.IterableDiffers)); };
IgxGridActionsBaseDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxGridActionsBaseDirective, selectors: [["", "igxGridActionsBase", ""]], viewQuery: function IgxGridActionsBaseDirective_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(IgxGridActionButtonComponent, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.buttons = _t);
    } }, inputs: { asMenuItems: "asMenuItems" } });
IgxGridActionsBaseDirective.ctorParameters = () => [
    { type: IgxIconService },
    { type: IterableDiffers }
];
IgxGridActionsBaseDirective.propDecorators = {
    buttons: [{ type: ViewChildren, args: [IgxGridActionButtonComponent,] }],
    asMenuItems: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridActionsBaseDirective, [{
        type: Directive,
        args: [{
                selector: '[igxGridActionsBase]'
            }]
    }], function () { return [{ type: ɵngcc1.IgxIconService }, { type: ɵngcc0.IterableDiffers }]; }, { asMenuItems: [{
            type: Input
        }], buttons: [{
            type: ViewChildren,
            args: [IgxGridActionButtonComponent]
        }] }); })();
