// igniteui-angular/esm2015/lib/action-strip/grid-actions/grid-pinning-actions.component.js
import { Component, HostBinding } from '@angular/core';
import { IgxGridActionsBaseDirective } from './grid-actions-base.directive';
import { pinLeft, unpinLeft, jumpDown, jumpUp } from '@igniteui/material-icons-extended';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from './grid-action-button.component';

function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_1_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 5);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_1_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r6); const ctx_r5 = ɵngcc0.ɵɵnextContext(2); return ctx_r5.scrollToRow($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r1.asMenuItems)("labelText", ctx_r1.grid.resourceStrings.igx_grid_actions_jumpDown_label);
} }
function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r8 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 6);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_2_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r8); const ctx_r7 = ɵngcc0.ɵɵnextContext(2); return ctx_r7.scrollToRow($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r2.asMenuItems)("labelText", ctx_r2.grid.resourceStrings.igx_grid_actions_jumpUp_label);
} }
function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_3_Template(rf, ctx) { if (rf & 1) {
    const _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 7);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_3_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); const ctx_r9 = ɵngcc0.ɵɵnextContext(2); return ctx_r9.pin($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r3.asMenuItems)("labelText", ctx_r3.grid.resourceStrings.igx_grid_actions_pin_label);
} }
function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_4_Template(rf, ctx) { if (rf & 1) {
    const _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "igx-grid-action-button", 8);
    ɵngcc0.ɵɵlistener("onActionClick", function IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_4_Template_igx_grid_action_button_onActionClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); const ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.unpin($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("asMenuItem", ctx_r4.asMenuItems)("labelText", ctx_r4.grid.resourceStrings.igx_grid_actions_unpin_label);
} }
function IgxGridPinningActionsComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_1_Template, 1, 2, "igx-grid-action-button", 1);
    ɵngcc0.ɵɵtemplate(2, IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_2_Template, 1, 2, "igx-grid-action-button", 2);
    ɵngcc0.ɵɵtemplate(3, IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_3_Template, 1, 2, "igx-grid-action-button", 3);
    ɵngcc0.ɵɵtemplate(4, IgxGridPinningActionsComponent_ng_container_0_igx_grid_action_button_4_Template, 1, 2, "igx-grid-action-button", 4);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.inPinnedArea && ctx_r0.pinnedTop);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.inPinnedArea && !ctx_r0.pinnedTop);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.pinned);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.pinned);
} }
export class IgxGridPinningActionsComponent extends IgxGridActionsBaseDirective {
    constructor() {
        super(...arguments);
        /**
         * Host `class.igx-action-strip` binding.
         * @hidden
         * @internal
         */
        this.cssClass = 'igx-action-strip__pinning-actions';
        this.iconsRendered = false;
    }
    /**
     * Getter to know if the row is pinned
     * @hidden
     * @internal
     */
    get pinned() {
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const context = this.strip.context;
        if (context && !this.iconsRendered) {
            this.registerSVGIcons();
            this.iconsRendered = true;
        }
        return context && context.pinned;
    }
    /**
     * Getter to know if the row is in pinned and ghost
     * @hidden
     * @internal
     */
    get inPinnedArea() {
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const context = this.strip.context;
        return this.pinned && !context.disabled;
    }
    /**
     * Getter to know if the row pinning is set to top or bottom
     * @hidden
     * @internal
     */
    get pinnedTop() {
        if (!this.isRow(this.strip.context)) {
            return;
        }
        return this.strip.context.grid.isRowPinningToTop;
    }
    /**
     * Pin the row according to the context.
     * @example
     * ```typescript
     * this.gridPinningActions.pin();
     * ```
     */
    pin(event) {
        if (event) {
            event.stopPropagation();
        }
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const row = this.strip.context;
        const grid = row.grid;
        grid.pinRow(row.rowID);
        this.strip.hide();
    }
    /**
     * Unpin the row according to the context.
     * @example
     * ```typescript
     * this.gridPinningActions.unpin();
     * ```
     */
    unpin(event) {
        if (event) {
            event.stopPropagation();
        }
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const row = this.strip.context;
        const grid = row.grid;
        grid.unpinRow(row.rowID);
        this.strip.hide();
    }
    scrollToRow(event) {
        if (event) {
            event.stopPropagation();
        }
        const context = this.strip.context;
        const grid = context.grid;
        grid.scrollTo(context.rowData, 0);
        this.strip.hide();
    }
    registerSVGIcons() {
        if (!this.isRow(this.strip.context)) {
            return;
        }
        const context = this.strip.context;
        const grid = context.grid;
        if (grid) {
            this.iconService.addSvgIconFromText(pinLeft.name, pinLeft.value, 'imx-icons');
            this.iconService.addSvgIconFromText(unpinLeft.name, unpinLeft.value, 'imx-icons');
            this.iconService.addSvgIconFromText(jumpDown.name, jumpDown.value, 'imx-icons');
            this.iconService.addSvgIconFromText(jumpUp.name, jumpDown.value, 'imx-icons');
        }
    }
}
IgxGridPinningActionsComponent.ɵfac = function IgxGridPinningActionsComponent_Factory(t) { return ɵIgxGridPinningActionsComponent_BaseFactory(t || IgxGridPinningActionsComponent); };
IgxGridPinningActionsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: IgxGridPinningActionsComponent, selectors: [["igx-grid-pinning-actions"]], hostVars: 2, hostBindings: function IgxGridPinningActionsComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-action-strip__pinning-actions", ctx.cssClass);
    } }, features: [ɵngcc0.ɵɵProvidersFeature([{ provide: IgxGridActionsBaseDirective, useExisting: IgxGridPinningActionsComponent }]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [[4, "ngIf"], ["iconSet", "imx-icons", "iconName", "jump-down", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconSet", "imx-icons", "iconName", "jump-up", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconSet", "imx-icons", "iconName", "pin-left", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconSet", "imx-icons", "iconName", "unpin-left", 3, "asMenuItem", "labelText", "onActionClick", 4, "ngIf"], ["iconSet", "imx-icons", "iconName", "jump-down", 3, "asMenuItem", "labelText", "onActionClick"], ["iconSet", "imx-icons", "iconName", "jump-up", 3, "asMenuItem", "labelText", "onActionClick"], ["iconSet", "imx-icons", "iconName", "pin-left", 3, "asMenuItem", "labelText", "onActionClick"], ["iconSet", "imx-icons", "iconName", "unpin-left", 3, "asMenuItem", "labelText", "onActionClick"]], template: function IgxGridPinningActionsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, IgxGridPinningActionsComponent_ng_container_0_Template, 5, 4, "ng-container", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.isRowContext);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.IgxGridActionButtonComponent], encapsulation: 2 });
IgxGridPinningActionsComponent.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-action-strip__pinning-actions',] }]
};
const ɵIgxGridPinningActionsComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(IgxGridPinningActionsComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridPinningActionsComponent, [{
        type: Component,
        args: [{
                selector: 'igx-grid-pinning-actions',
                template: "<ng-container *ngIf=\"isRowContext\">\n    <igx-grid-action-button *ngIf=\"inPinnedArea && pinnedTop\" [asMenuItem]=\"asMenuItems\" iconSet=\"imx-icons\" iconName=\"jump-down\" [labelText]=\"grid.resourceStrings.igx_grid_actions_jumpDown_label\" (onActionClick)=\"scrollToRow($event)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"inPinnedArea && !pinnedTop\" [asMenuItem]=\"asMenuItems\" iconSet=\"imx-icons\" iconName=\"jump-up\" [labelText]=\"grid.resourceStrings.igx_grid_actions_jumpUp_label\" (onActionClick)=\"scrollToRow($event)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"!pinned\" [asMenuItem]=\"asMenuItems\" iconSet=\"imx-icons\" iconName=\"pin-left\" [labelText]=\"grid.resourceStrings.igx_grid_actions_pin_label\" (onActionClick)=\"pin($event)\"></igx-grid-action-button>\n    <igx-grid-action-button *ngIf=\"pinned\" [asMenuItem]=\"asMenuItems\" iconSet=\"imx-icons\" iconName=\"unpin-left\" [labelText]=\"grid.resourceStrings.igx_grid_actions_unpin_label\" (onActionClick)=\"unpin($event)\"></igx-grid-action-button>\n</ng-container>",
                providers: [{ provide: IgxGridActionsBaseDirective, useExisting: IgxGridPinningActionsComponent }]
            }]
    }], null, { cssClass: [{
            type: HostBinding,
            args: ['class.igx-action-strip__pinning-actions']
        }] }); })();
