// igniteui-angular/esm2015/lib/expansion-panel/expansion-panel.directives.js
import { Directive, HostBinding } from '@angular/core';
/**
 * @hidden @internal
 */
import * as ɵngcc0 from '@angular/core';
export class IgxExpansionPanelTitleDirective {
    constructor() {
        this.cssClass = `igx-expansion-panel__header-title`;
    }
}
IgxExpansionPanelTitleDirective.ɵfac = function IgxExpansionPanelTitleDirective_Factory(t) { return new (t || IgxExpansionPanelTitleDirective)(); };
IgxExpansionPanelTitleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxExpansionPanelTitleDirective, selectors: [["igx-expansion-panel-title"]], hostVars: 2, hostBindings: function IgxExpansionPanelTitleDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-expansion-panel__header-title", ctx.cssClass);
    } } });
IgxExpansionPanelTitleDirective.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-expansion-panel__header-title',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExpansionPanelTitleDirective, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: 'igx-expansion-panel-title'
            }]
    }], function () { return []; }, { cssClass: [{
            type: HostBinding,
            args: ['class.igx-expansion-panel__header-title']
        }] }); })();
/**
 * @hidden @internal
 */
export class IgxExpansionPanelDescriptionDirective {
    constructor() {
        this.cssClass = `igx-expansion-panel__header-description`;
    }
}
IgxExpansionPanelDescriptionDirective.ɵfac = function IgxExpansionPanelDescriptionDirective_Factory(t) { return new (t || IgxExpansionPanelDescriptionDirective)(); };
IgxExpansionPanelDescriptionDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxExpansionPanelDescriptionDirective, selectors: [["igx-expansion-panel-description"]], hostVars: 2, hostBindings: function IgxExpansionPanelDescriptionDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("igx-expansion-panel__header-description", ctx.cssClass);
    } } });
IgxExpansionPanelDescriptionDirective.propDecorators = {
    cssClass: [{ type: HostBinding, args: ['class.igx-expansion-panel__header-description',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExpansionPanelDescriptionDirective, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: 'igx-expansion-panel-description'
            }]
    }], function () { return []; }, { cssClass: [{
            type: HostBinding,
            args: ['class.igx-expansion-panel__header-description']
        }] }); })();
/**
 * @hidden @internal
 */
export class IgxExpansionPanelIconDirective {
}
IgxExpansionPanelIconDirective.ɵfac = function IgxExpansionPanelIconDirective_Factory(t) { return new (t || IgxExpansionPanelIconDirective)(); };
IgxExpansionPanelIconDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxExpansionPanelIconDirective, selectors: [["igx-expansion-panel-icon"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxExpansionPanelIconDirective, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: 'igx-expansion-panel-icon'
            }]
    }], null, null); })();
