// igniteui-angular/esm2015/lib/grids/hiding/column-hiding-item.directive.js
import { Directive } from '@angular/core';
import { ColumnChooserItemBaseDirective } from '../column-chooser-item-base';
/** @hidden */
import * as ɵngcc0 from '@angular/core';
export class IgxColumnHidingItemDirective extends ColumnChooserItemBaseDirective {
    constructor() {
        super('hidden');
    }
    get disabled() {
        return this.column.disableHiding;
    }
}
IgxColumnHidingItemDirective.ɵfac = function IgxColumnHidingItemDirective_Factory(t) { return new (t || IgxColumnHidingItemDirective)(); };
IgxColumnHidingItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxColumnHidingItemDirective, selectors: [["", "igxColumnHidingItem", ""]], features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
IgxColumnHidingItemDirective.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnHidingItemDirective, [{
        type: Directive,
        args: [{
                selector: '[igxColumnHidingItem]'
            }]
    }], function () { return []; }, null); })();
