// igniteui-angular/esm2015/lib/grids/hiding/hiding.module.js
import { NgModule } from '@angular/core';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnHidingItemDirective } from './column-hiding-item.directive';
import { IgxColumnHidingComponent } from './column-hiding.component';
import * as ɵngcc0 from '@angular/core';
export class IgxColumnHidingModule {
}
IgxColumnHidingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxColumnHidingModule });
IgxColumnHidingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxColumnHidingModule_Factory(t) { return new (t || IgxColumnHidingModule)(); }, imports: [[
            IgxGridSharedModules
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxColumnHidingModule, { declarations: function () { return [IgxColumnHidingComponent,
        IgxColumnHidingItemDirective]; }, imports: function () { return [IgxGridSharedModules]; }, exports: function () { return [IgxColumnHidingComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnHidingModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxColumnHidingComponent,
                    IgxColumnHidingItemDirective
                ],
                imports: [
                    IgxGridSharedModules
                ],
                exports: [
                    IgxColumnHidingComponent
                ]
            }]
    }], null, null); })();
