// igniteui-angular/esm2015/lib/grids/toolbar/toolbar.directive.js
import { Directive, TemplateRef } from '@angular/core';
/**
 * The IgxGridToolbarCustomContentDirective directive is used to mark an 'ng-template' (with
 * the 'igxToolbarCustomContent' selector) defined in the IgxGrid which is used to provide
 * custom content for cener part of the IgxGridToolbar.
 */
import * as ɵngcc0 from '@angular/core';
export class IgxGridToolbarCustomContentDirective {
    constructor(template) {
        this.template = template;
    }
}
IgxGridToolbarCustomContentDirective.ɵfac = function IgxGridToolbarCustomContentDirective_Factory(t) { return new (t || IgxGridToolbarCustomContentDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef)); };
IgxGridToolbarCustomContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxGridToolbarCustomContentDirective, selectors: [["", "igxToolbarCustomContent", ""]] });
IgxGridToolbarCustomContentDirective.ctorParameters = () => [
    { type: TemplateRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridToolbarCustomContentDirective, [{
        type: Directive,
        args: [{
                selector: '[igxToolbarCustomContent]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef }]; }, null); })();
