// igniteui-angular/esm2015/lib/grids/toolbar/toolbar.module.js
import { NgModule } from '@angular/core';
import { IgxGridToolbarComponent } from './grid-toolbar.component';
import { IgxGridToolbarCustomContentDirective } from './toolbar.directive';
import { IgxGridSharedModules } from '../common/shared.module';
import { IgxColumnHidingModule } from '../hiding/hiding.module';
import { IgxColumnPinningModule } from '../pinning/pinning.module';
import { IgxColumnActionsModule } from '../column-actions/column-actions.module';
import * as ɵngcc0 from '@angular/core';
export class IgxGridToolbarModule {
}
IgxGridToolbarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridToolbarModule });
IgxGridToolbarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridToolbarModule_Factory(t) { return new (t || IgxGridToolbarModule)(); }, imports: [[
            IgxGridSharedModules,
            IgxColumnHidingModule,
            IgxColumnPinningModule,
            IgxColumnActionsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridToolbarModule, { declarations: function () { return [IgxGridToolbarComponent,
        IgxGridToolbarCustomContentDirective]; }, imports: function () { return [IgxGridSharedModules,
        IgxColumnHidingModule,
        IgxColumnPinningModule,
        IgxColumnActionsModule]; }, exports: function () { return [IgxGridToolbarComponent,
        IgxGridToolbarCustomContentDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridToolbarModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxGridToolbarComponent,
                    IgxGridToolbarCustomContentDirective
                ],
                imports: [
                    IgxGridSharedModules,
                    IgxColumnHidingModule,
                    IgxColumnPinningModule,
                    IgxColumnActionsModule
                ],
                exports: [
                    IgxGridToolbarComponent,
                    IgxGridToolbarCustomContentDirective
                ]
            }]
    }], null, null); })();
