// igniteui-angular/esm2015/lib/grids/moving/moving.module.js
import { NgModule } from '@angular/core';
import { IgxColumnMovingDragDirective } from './moving.drag.directive';
import { IgxColumnMovingDropDirective } from './moving.drop.directive';
import * as ɵngcc0 from '@angular/core';
export class IgxColumnMovingModule {
}
IgxColumnMovingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxColumnMovingModule });
IgxColumnMovingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxColumnMovingModule_Factory(t) { return new (t || IgxColumnMovingModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxColumnMovingModule, { declarations: function () { return [IgxColumnMovingDropDirective,
        IgxColumnMovingDragDirective]; }, exports: function () { return [IgxColumnMovingDropDirective,
        IgxColumnMovingDragDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnMovingModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxColumnMovingDropDirective,
                    IgxColumnMovingDragDirective
                ],
                exports: [
                    IgxColumnMovingDropDirective,
                    IgxColumnMovingDragDirective
                ]
            }]
    }], null, null); })();
