// igniteui-angular/esm2015/lib/grids/filtering/base/filtering.module.js
import { NgModule } from '@angular/core';
import { IgxGridFilteringCellComponent } from './grid-filtering-cell.component';
import { IgxGridFilteringRowComponent } from './grid-filtering-row.component';
import { IgxGridSharedModules } from '../../common/shared.module';
import { IgxGridPipesModule } from '../../common/grid-pipes.module';
import * as ɵngcc0 from '@angular/core';
export class IgxGridFilteringModule {
}
IgxGridFilteringModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxGridFilteringModule });
IgxGridFilteringModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxGridFilteringModule_Factory(t) { return new (t || IgxGridFilteringModule)(); }, imports: [[
            IgxGridSharedModules,
            IgxGridPipesModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxGridFilteringModule, { declarations: function () { return [IgxGridFilteringCellComponent,
        IgxGridFilteringRowComponent]; }, imports: function () { return [IgxGridSharedModules,
        IgxGridPipesModule]; }, exports: function () { return [IgxGridFilteringCellComponent,
        IgxGridFilteringRowComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxGridFilteringModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxGridFilteringCellComponent,
                    IgxGridFilteringRowComponent
                ],
                imports: [
                    IgxGridSharedModules,
                    IgxGridPipesModule
                ],
                exports: [
                    IgxGridFilteringCellComponent,
                    IgxGridFilteringRowComponent
                ]
            }]
    }], null, null); })();
