// igniteui-angular/esm2015/lib/grids/pinning/pinning.module.js
import { NgModule } from '@angular/core';
import { IgxColumnPinningItemDirective } from './pinning.directive';
import { IgxColumnPinningComponent } from './column-pinning.component';
import { IgxGridSharedModules } from '../common/shared.module';
import * as ɵngcc0 from '@angular/core';
export class IgxColumnPinningModule {
}
IgxColumnPinningModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IgxColumnPinningModule });
IgxColumnPinningModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IgxColumnPinningModule_Factory(t) { return new (t || IgxColumnPinningModule)(); }, imports: [[
            IgxGridSharedModules
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IgxColumnPinningModule, { declarations: function () { return [IgxColumnPinningItemDirective,
        IgxColumnPinningComponent]; }, imports: function () { return [IgxGridSharedModules]; }, exports: function () { return [IgxColumnPinningComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnPinningModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    IgxColumnPinningItemDirective,
                    IgxColumnPinningComponent
                ],
                imports: [
                    IgxGridSharedModules
                ],
                exports: [
                    IgxColumnPinningComponent
                ]
            }]
    }], null, null); })();
