// igniteui-angular/esm2015/lib/grids/pinning/pinning.directive.js
import { Directive } from '@angular/core';
import { ColumnChooserItemBaseDirective } from '../column-chooser-item-base';
import * as ɵngcc0 from '@angular/core';
export class IgxColumnPinningItemDirective extends ColumnChooserItemBaseDirective {
    constructor() {
        super('pinned');
    }
    /**
     * Returns whether a column could be pinned.
     * It's not possible to pin a column if there is not enough space for the unpinned area.
     * ```typescript
     * const columnItem: IgxColumnPinningItemDirective;
     * this.columnItem.pinnable;
     * ```
     */
    get pinnable() {
        return this.column.pinnable;
    }
    /**
     * Returns whether a column have pinning enabled
     */
    get pinningDisabled() {
        return this.column.disablePinning;
    }
}
IgxColumnPinningItemDirective.ɵfac = function IgxColumnPinningItemDirective_Factory(t) { return new (t || IgxColumnPinningItemDirective)(); };
IgxColumnPinningItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: IgxColumnPinningItemDirective, selectors: [["", "igxColumnPinningItem", ""]], features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
IgxColumnPinningItemDirective.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IgxColumnPinningItemDirective, [{
        type: Directive,
        args: [{
                selector: '[igxColumnPinningItem]'
            }]
    }], function () { return []; }, null); })();
